The advent of next-generation sequencing has brought forth an era where datasets containing genotypic information for thousands of individuals are common. The key to leveraging rich datasets to generate impactful biomedical insights are high-quality computational tools for biological data analysis. The field of population genetics has a long tradition of using mathematical models to investigate how evolutionary forces shape the dynamics of genetic variants and their biological implications. More recently, artificial intelligence and machine learning (AI/ML) methods have demonstrated state-of-the-art performance for a wide range of applications involving big data and are also increasingly recognized as a powerful tool for population-genetic research. My thesis research addresses the unique promises and challenges of analyzing genomic data with AI/ML methods by developing rigorous, scalable and innovative deep learning models for population-genetic inference tasks.

A fundamental pursuit in evolutionary genetics is to identify beneficial mutations and measure the strength of their selective advantage, based on patterns of genetic variation across populations. Studies of positively-selected loci have led to new insights into the molecular genetic roles or disease relevance of particular genomic elements. Despite many advances, major limitations remain in the sensitivity and accuracy of computational methods for identifying and characterizing selection. These limitations stem, in part, from the difficulty of estimating selective effects directly from DNA sequences. We developed a novel deep-learning method called SIA (\textit{S}election \textit{I}nference using the \textit{A}RG), which makes use of a rich set of features extracted from a reconstructed ancestral recombination graph (ARG) to make accurate inferences about selection from large-scale genomic data. The ARG augments the raw sequences by encoding their complete evolutionary history. By exploiting both the richness of information in the ARG and the flexibility and scalability of deep-learning models, SIA offers exceptional prediction performance, exceeding that of many classes of recently published methods.

An interesting feature of the new generation of AI/ML methods for applications in population genetics, including SIA, is that they generally rely on simulated data for supervised training. This simulate-and-train paradigm has the advantage of virtually unlimited training data that is perfectly labeled, but the disadvantage that its performance depends strongly on modeling assumptions for simulations and can fail when the simulations are badly mis-specified. To go beyond the current ad-hoc methods for handling this essential problem, we devised a domain-adaptive framework for deep-learning models trained on simulated population genetic data. We used domain adaptation – a specific form of transfer learning – to train models on one data distribution (simulated genomic data) that can perform well when applied to datasets drawn from a different distribution (real genomic data). Our framework effectively addresses the simulation mis-specification problem which has been the major concern about current applications of AI/ML approaches in population genetics.

The deep-learning frameworks we have developed so far mark a pivotal step to capitalize on the momentum of technological progress in sequencing, computing capacity as well as AI/ML algorithms, but only the beginning of deep-learning approaches to evolutionary modeling. Recently, large language models (LLMs) of protein and DNA have shown promising performance in a variety of problems in molecular biology such as protein structure or variant effect prediction. Similarly, large generative pre-trained evolutionary models based on genealogical embeddings of the ARG in the future have the potential to revolutionize population genetic research. Such models can be trained in a self-supervised manner with an incredibly wide range of simulations to learn a generative model of many evolutionary processes, which in principle can be fine-tuned to perform diverse tasks such as inference of demography, population structure or admixture events. Hence, we envision that our work will ultimately open the door to many more AL/ML methods tailored to population genetic inference.
