\chapter{Domain-adaptive neural networks improve supervised machine learning based on simulated population genetic data} \label{chapter4}

\textit{Content of this chapter was previously uploaded to bioRxiv (2023) under the title ``Domain-adaptive neural networks improve supervised machine learning based on simulated population genetic data" by Ziyi Mo and Adam Siepel. The manuscript was published in PLoS Genetics (2023) under the same title.}

\section{Abstract}
Investigators have recently introduced powerful methods for population genetic inference that rely on supervised machine learning from simulated data. Despite their performance advantages, these methods can fail when the simulated training data does not adequately resemble data from the real world. Here, we show that this “simulation mis-specification” problem can be framed as a “domain adaptation” problem, where a model learned from one data distribution is applied to a dataset drawn from a different distribution. By applying an established domain-adaptation technique based on \iac{GRL}, originally introduced for image classification, we show that the effects of simulation mis-specification can be substantially mitigated. We focus our analysis on two state-of-the-art deep-learning population genetic methods—\ac{SIA}, which infers positive selection from features of the \acf{ARG}, and ReLERNN, which infers recombination rates from genotype matrices. In the case of \ac{SIA}, the domain adaptive framework also compensates for \ac{ARG} inference error. Using the \ac{dadaSIA} model, we estimate improved selection coefficients at selected loci in the 1000 Genomes CEU population. We anticipate that domain adaptation will prove to be widely applicable in the growing use of supervised machine learning in population genetics.

\section{Introduction}
Advances in genome sequencing have allowed population genetic analyses to be applied to many thousands of individual genome sequences (\cite{auton_global_2015,sudlow_uk_2015,karczewski_mutational_2020}). Given adequately rigorous and scalable computational tools for analysis, these rich catalogs of genetic variation provide opportunities for addressing many important questions in areas such as human evolution, plant genetics, and the ecology of non-model organisms. Deep-learning methods, already well-established in other application areas (\cite{lecun_deep_2015}), have proven to be good matches for these analytical tasks and have recently been successfully applied to many problems in population genetics (\cite{sheehan_deep_2016,kern_diploshic_2018,schrider_supervised_2018,flagel_unreasonable_2019,torada_imagene_2019,adrion_predicting_2020,caldas_inference_2022,hejase_deep-learning_2022,korfmann_deep_2023,huang_harnessing_2023}).

The key to the success of deep learning in population genetics has been the use of large amounts of simulated data for training. Under simplifying, yet largely realistic, assumptions, evolution plays by relatively straightforward rules. By exploiting these rules and advances in computing power, a new generation of computational simulators has made it possible to efficiently produce large quantities of perfectly labeled synthetic data across a wide range of evolutionary scenarios (\cite{haller_tree-sequence_2019,haller_slim_2019,baumdicker_efficient_2022}). At the same time, programming libraries such as stdpopsim have made these simulators accessible to a broad community of researchers while improving the reproducibility of simulation workflows (\cite{adrion_community-maintained_2020,lauterbur_expanding_2022}). The facility of generating synthetic training data serves as the foundation of the new simulate-and-train paradigm of supervised machine learning for population genetics inference (Fig. \ref{fig:DA-F1}A; \cite{schrider_supervised_2018,korfmann_deep_2023,huang_harnessing_2023}).

\begin{figure}[h]
    \centering
    \includegraphics[width=\textwidth]{DA_figs/DA_F1.PNG}
    \caption[Unsupervised domain adaptation in the context of population genetic inference.]{\textbf{Unsupervised domain adaptation in the context of population genetic inference.} \textbf{A)} A high-level overview of the supervised machine-learning approach for population genetic inference and how domain adaptation fits into the paradigm. \textbf{B)} Example formulations of the unsupervised domain adaptation problem with application to computer vision and population genetics. Note that in the specific case of \ac{SIA}, which uses features of the \ac{ARG}, the source domain data always consist of true genealogies generated in simulations, whereas the target domain data always consist of inferred genealogies reconstructed from observed sequence data. \textbf{C)} Four benchmarking scenarios considered in this study. The original model was both trained and tested on source domain data (simulation benchmark), both trained and tested on target domain data (hypothetical true model), or trained on source domain data but applied to target domain data (standard model application). These three cases contextualize the performance of the domain-adaptive model (see \nameref{DA-methods} for details). Gold squares represent source domain data, blue circles represent target domain data and crosses ($\mathbf{\times}$) represent labels.}
    \label{fig:DA-F1}
\end{figure}

At the same time, this paradigm is highly dependent on well-specified models for simulation (\cite{korfmann_deep_2023}). If the simulation assumptions do not match the underlying generative process of the real data -- that is, in the presence of \textit{simulation mis-specification} -- the trained deep-learning model may reflect the biases in the simulated data and perform poorly on real data. Indeed, previous studies have shown that, despite being robust to mild to moderate levels of mis-specification, performance inevitably degrades when the mismatch becomes severe (\cite{adrion_predicting_2020,hejase_deep-learning_2022}).

In a typical workflow, key simulation parameters such as the mutation rate, recombination rate, and parameters of the demographic model are either estimated from the data or obtained from the literature (Fig. \ref{fig:DA-F1}A; \cite{adrion_community-maintained_2020,lauterbur_expanding_2022}). Sometimes these parameters are allowed to vary during simulation, and sometimes investigators evaluate the sensitivity of predictions to departures from the assumed range, but there is typically no way to ensure that the ranges considered are adequately large. Moreover, these benchmarks do not usually account for under-parameterization of the demographic model. Particularly in the case of non-model organisms, the quality of the estimates can be further limited by the availability of data. Overall, some degree of mis-specification in the simulated training data is impossible to avoid.

One way to mitigate the effects of simulation mis-specification would be to engineer a simulator to force the simulated data to be compatible with real data. For example, one could simulate from an overdispersed distribution of parameters followed by a rejection sampling step (based on summary statistics) as in \acf{ABC} methods, or one could use a \ac{GAN} (\cite{wang_automatic_2021}) to mimic the real data. These methods tend to be costly, however. For example, \ac{ABC} methods scale poorly with the dimensionality of the parameter space, and \acp{GAN} are notoriously hard to train.

Here we consider the alternative approach of adopting a deep-learning model that is explicitly designed to account for and mitigate the mismatch between simulated and real data (Fig. \ref{fig:DA-F1}A). A standard machine learning model aims to make accurate predictions on data following the same probability distribution as the training instances. In contrast, the task of building well-performing models for a target dataset that has a \textit{different} distribution from the training dataset is termed “domain adaptation” in the machine-learning literature (\cite{csurka_comprehensive_2017,wilson_survey_2020}). A typical setting of interest for domain adaptation is image classification (Fig. \ref{fig:DA-F1}B). For example, suppose a digit-recognition model is needed for the Street View House Numbers (SVHN) dataset (the “target domain”), but abundant labeled training data is only available from the MNIST dataset of handwritten digits (the “source domain”). In this case, a method needs to train on one dataset and perform well on another, despite systematic differences between the two data distributions.

Various strategies for domain adaptation have been introduced. Prior to the advent of deep learning, early methods focused on reweighting training instances according to their likelihoods of being a source or target example (\cite{shimodaira_improving_2000,dai_boosting_2007}) or explicitly manipulating a feature space through augmentation (\cite{daume_iii_frustratingly_2009}), alignment (\cite{fernando_unsupervised_2013,sun_return_2016}) or transformation (\cite{pan_domain_2011}). Recently, specialized neural network architectures have been developed for deep domain adaptation. Most model architectures of this kind share the common goal of learning a “domain-invariant” representation of the data through a feature extractor neural network, for example, by minimizing domain divergence (\cite{rozantsev_beyond_2019}), by adversarial training (\cite{ganin_unsupervised_2014,liu_coupled_2016}) or through an auxiliary reconstruction task (\cite{ghifary_deep_2016}). Domain adaptation so far has been most widely applied in the fields of computer vision (e.g., using stock photos for semantic segmentation of real photos) and natural language processing (e.g., using Amazon product reviews for sentiment analysis of movies and TV shows) where large, heterogeneous datasets are common but producing labeled training examples can be labor intensive (\cite{wilson_survey_2020}). More recently, deep domain adaptation has been used in regulatory genomics to enable cross-species transcription-factor-binding-site prediction (\cite{cochran_domain-adaptive_2022}).

In this work, we reframe the simulation mis-specification problem in population genetics as an unsupervised domain adaptation problem -- unsupervised in the sense that data from the target domain is not labeled (Fig. \ref{fig:DA-F1}B). In particular, we use population-genetic simulations to obtain large amounts of perfectly labeled training data in the source domain. We then seek to apply the trained model to unlabeled real data in the target domain. We use domain adaptation techniques to explicitly account for the mismatch between these two domains when training the model.

To demonstrate the feasibility of this approach, we incorporated a domain-adaptive neural network architecture into two published deep learning models for population genetic inference: 1) SIA (\cite{hejase_deep-learning_2022}), which identifies selective sweeps based on the \acf{ARG}, and 2) ReLERNN (\cite{adrion_predicting_2020}), which infers recombination rates from raw genotypic data. Through extensive simulation studies, we demonstrated that the domain adaptive versions of the models significantly outperformed the standard versions under realistic scenarios of simulation mis-specification. Our domain-adaptive framework for utilizing mis-specified synthetic data for supervised learning opens the door to many more applications in population genetics.

\section{Results}

\subsection{Experimental design}
We created domain-adaptive versions of the \ac{SIA} and ReLERNN models, each of which employed a \acf{GRL} (\cite{ganin_unsupervised_2014}) (Fig. \ref{fig:DA-F2}A\&B). As noted, the goal of domain adaptation is to establish a “domain-invariant” representation of the data (Fig. \ref{fig:DA-F1}A). Our neural networks consist of two major components: the original networks (“feature extractor” in green and “label predictor” in blue in Fig. \ref{fig:DA-F2}A\&B), which are applied only to labeled examples from the “source” (simulated) domain; and alternative branches (“domain classifier” in yellow in Fig. \ref{fig:DA-F2}A\&B), which use the same feature-extraction portions of the first networks but have the distinct goal of distinguishing data from the “source” (simulated) and “target” (real) domains (they are applied to both). When the neural network is trained by back-propagation, the \ac{GRL} reverses the sign of the gradient for the feature extractor with respect to the domain-classifier loss. By doing so, the \ac{GRL} systematically undermines this secondary goal of distinguishing the two domains (Fig. \ref{fig:DA-F2}, see \nameref{DA-methods} for details), and therefore promotes domain invariance in feature extraction.

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{DA_figs/DA_F2.PNG}
    \caption[Neural network architecture for domain adaptation.]{\textbf{Neural network architecture for domain adaptation.} The model architectures incorporating \acfp{GRL} for \textbf{A)} \ac{SIA} and \textbf{B)} ReLERNN. The feature extractor of \ac{SIA} contains $1.49 \times 10^5$ trainable parameters, whereas the label predictor and domain classifier contains $1.22 \times 10^8$ each. The feature extractor of ReLERNN contains $1.52 \times 10^6$ trainable parameters, whereas the label predictor and domain classifier contains $1.49 \times 10^5$ each. Note that the total number of trainable parameters includes those in batch normalization layers. \textbf{C)} When training the networks, each minibatch of training data consists of two components: (1) labeled data from the source domain fed through the feature extractor and the label predictor; and (2) a mixture of unlabeled data from both the source and target domains fed through the feature extractor and the domain classifier. The first component trains the model to perform its designated task. However, the \ac{GRL} inverts the loss function for the second component, discouraging the model from differentiating the two domains and leading to the extraction of “domain-invariant” features.}
    \label{fig:DA-F2}
\end{figure}

We designed two sets of benchmark experiments to assess the performance of the domain-adaptive models relative to the standard models. In both cases, we tested the methods using “real” data in the target domain that was actually generated by simulation, but included features not considered by the simpler simulator used for the source domain. In the first set of experiments, background selection was present in the target domain but not the source domain. In the second set of experiments, the demographic model used for the source-domain simulations was estimated from “real” data generated under a more complex demographic model and was therefore somewhat mis-specified (as detailed below). Below we refer to these as the “background selection” and “demography mis-specification” experiments.

\subsection{Performance of domain-adaptive \ac{SIA} model}
We compared the performance of the \acf{dadaSIA} model to that of the standard \ac{SIA} model on held-out “real” data, considering both a classification (distinguishing selective sweeps from neutrality) and a regression (inferring selection coefficients) task. In all cases, we focused on a comparison of the domain-adaptive model to the standard case where a model is simply trained on data from the source domain and then applied to the target domain (“standard model”; Fig. \ref{fig:DA-F1}C). Note that the version of \ac{SIA} used by both the domain-adaptive and standard models includes a variety of minor improvements that led to modest gains in performance over the previously published version (see \textit{Updates to genealogical features and deep learning architecture for the \ac{SIA} model} in \nameref{DA-methods} and Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S1B\&C} in Appendix \ref{appxC}). The codebase of the original \ac{SIA} model has been updated accordingly.

For additional context, we also considered the two cases where the training and testing domains matched (“source-matched” or “target-matched”; Fig. \ref{fig:DA-F1}C)—although we note that these cases are not achievable with real data and provide only hypothetical upper bounds on performance. Notably, in the source-matched (or “simulation benchmark”) case, the standard model is both trained and tested with true genealogies from source-domain simulations. By contrast, in the target-matched (or “hypothetical true model”) case, the standard model is trained as if target-domain data with ground-truth selection coefficient labels were available. Since genealogies need to be inferred in the target domain (Fig. \ref{fig:DA-F1}B), the hypothetical true model is both trained and tested with inferred genealogies (see \textit{Setup of benchmarking experiments} in \nameref{DA-methods} for details).

As noted, we considered two types of mis-specification, background selection and demographic mis-specification. In the background selection experiments, the target domain experienced selection in a central “genic” region (following a \acsu{DFE} from \cite{boyko_assessing_2008}), leading to background selection in flanking regions. This genic region was omitted in the source domain. In the demographic mis-specification experiments, the demographic model for source-domain simulations was inferred from “real” data using G-PhoCS (\cite{gronau_bayesian_2011}). Both the real (target domain) and inferred (source domain) models assumed three populations with migration, but the inferred model was under-parameterized and its parameters differed substantially from the real model (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S1A} in Appendix \ref{appxC}) (see \nameref{DA-methods} for details).

In both the background selection and demography mis-specification experiments, and in both the classification and regression tasks, the domain-adaptive \ac{SIA} model substantially improved on the standard model (Fig. \ref{fig:DA-F3}). Indeed, in all cases, the domain-adaptive model (turquoise lines in Fig. \ref{fig:DA-F3}A\&C) nearly achieved the upper bound of the hypothetical true model (dashed gray lines) and clearly outperformed the standard model (gold lines), suggesting that domain adaptation had largely “rescued” \ac{SIA} from the effects of simulation mis-specification (see also Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S2C\&D} in Appendix \ref{appxC}). The standard model performed particularly poorly on the regression task (Fig. \ref{fig:DA-F3}B\&D), but the domain-adaptive model achieved substantial improvements, reducing both the absolute error as well as the upward bias of the estimation (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S2C\&D} in Appendix \ref{appxC}).

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{DA_figs/DA_F3.PNG}
    \caption[Performance of domain-adaptive \ac{SIA} models.]{\textbf{Performance of domain-adaptive \ac{SIA} models.} Results are shown from (\textbf{A}, \textbf{B}) the background-selection and (\textbf{C}, \textbf{D}) the demography-mis-specification experiments. (\textbf{A}, \textbf{C}) Precision-recall curves for sweep classification. (\textbf{B}, \textbf{D}) Contour plots summarizing true (horizontal axis) vs. inferred (vertical axis) selection coefficients ($s$) for the standard (gold) and domain adaptive (turquoise) models as evaluated on the held-out test dataset. The ridge along the horizontal axis of each contour is traced by a dashed line, representing the mode of the inferred value for each true value of $s$. Raw data underlying the contour plots are presented in Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032\#sec018}{S2} in Appendix \ref{appxC}. See Fig. \ref{fig:DA-F1}C for definition of the model labels.}
    % https://tex.stackexchange.com/questions/366909/and-other-characters-in-a-url
    \label{fig:DA-F3}
\end{figure}

The comparisons with the simulation benchmark and hypothetical true model were also informative in other ways. Notice that performance in the simulation benchmark case was considerably better than that in all other cases, including the hypothetical true model. For \ac{SIA} in particular, the \ac{ARG} is “known” (fixed in simulation) in the source domain, whereas in the target domain it must be inferred (Fig. \ref{fig:DA-F1}B). Thus, the difference between the simulation benchmark (source-matched) and hypothetical true model (target-matched) cases represents a rough measure of the importance of \ac{ARG} inference error (see \nameref{DA-discussion}). In addition, note that in many studies, benchmarking of population-genetic models is performed using the same, or similar, simulations as those used for training, as with our hypothetical true model. Thus, the difference between the hypothetical true model and the standard model is representative of the degree to which benchmarks of this kind may be overly optimistic about performance, depending on the degree to which the simulations are mis-specified.

We further investigated the effect of imbalanced training data from the target domain on the performance of the domain-adaptive model in the context of sweep classification. Despite the ability to simulate perfectly class-balanced labeled data in the source domain, in practice we have no control over whether real data are balanced. Using simulations for the background selection mis-specification experiments, we tested the performance of the domain-adaptive \ac{SIA} model classifying sweeps when trained with unlabeled “real” data under different proportions of sweep vs. neutral examples. While a balanced dataset yielded the best performance, significantly skewed datasets (20\% or 80\% sweep examples) still provided the domain-adaptive model with reasonable improvement upon the standard model (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032\#sec018}{S3A\&B} in Appendix \ref{appxC}). The exception appeared to be when the target domain data consisted entirely of sweep examples (100\% sweep). Although highly unrealistic, this scenario demonstrates that the domain-adaptive model can underperform the standard model when the target domain data follow a radically different distribution.

Another type of imbalance arises if only a limited amount of target domain data is available to train the domain-adaptive model. Using the same set of simulations for the background selection mis-specification experiments, we tested the performance of the domain-adaptive \ac{SIA} model when trained with less target domain data. With the target domain data at only 10\% of the source domain data (source:target ratio = 10:1), the model suffered a noticeable drop in performance yet still maintained a clear advantage over the standard model (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032\#sec018}{S3C-E} in Appendix \ref{appxC}). We did not examine the case where there is more target domain than source domain data, since one could always simulate additional source domain data to match the size of the target domain. In summary, our experiments suggest that domain adaptation can accommodate reduced or imbalanced data for the target domain but there is a cost in performance if the reduction or imbalance is extreme.

\subsection{Performance of domain-adaptive ReLERNN model}
We performed a parallel set of experiments with a domain-adaptive version of ReLERNN. In this case, the background selection experiment was essentially the same as for \ac{SIA}, but we used a simpler design for the demography mis-specification experiment, following \cite{adrion_predicting_2020}. Briefly, the “real” (target domain) data was generated according to the out-of-Africa European demographic model estimated by \cite{tennessen_evolution_2012}. By contrast, the simulated data for the source domain simply assumed a constant-sized panmictic population at equilibrium with $N_e=\frac{\hat{\theta}_W}{4\mu}$, where  $\hat{\theta}_W$ is the Watterson estimator obtained from the “real” data (see \nameref{DA-methods} for details).

Similar to our results for \ac{SIA}, the domain-adaptive ReLERNN model both reduced the \ac{MAE} and corrected for the downward bias in recombination-rate estimates compared to the standard model (Figs. \ref{fig:DA-F4} and \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S6} in Appendix \ref{appxC}). In the background-selection experiment, the standard ReLERNN model performed quite well (Figs. \ref{fig:DA-F4}A and \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S6A} in Appendix \ref{appxC}, $\mathrm{MAE} = 5.60\times 10^{-9}$), but the domain-adaptive ReLERNN model nonetheless further reduced the MAE to $4.41\times 10^{-9}$ (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S6C} in Appendix \ref{appxC}, Welch’s $t$-test: $n = 25,000$, $t =31.0$, $p<10^{-208}$). The advantage of the domain-adaptive model was more apparent in the demography-mis-specification experiment (Figs. \ref{fig:DA-F4}B and \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S6B} in Appendix \ref{appxC}), where it reduced the MAE from $8.06\times 10^{-9}$ to $5.45\times 10^{-9}$ (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S6D} in Appendix \ref{appxC}, Welch’s $t$-test, $n = 25,000$, $t = 72.4$, $p<10^{-323}$). Notably, our results for the standard model in the demography-mis-specification experiment were highly similar to those reported by \cite{adrion_predicting_2020}, including the approximate mean and range of the raw error (compare \href{https://academic.oup.com/view-large/figure/204168641/msaa038f4.tif}{Fig. 4A} from \cite{adrion_predicting_2020} and Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S6D} in Appendix \ref{appxC}), as well as the downward bias.

\begin{figure}[h]
    \centering
    \includegraphics[width=\textwidth]{DA_figs/DA_F4.PNG}
    \caption[Performance of domain-adaptive ReLERNN models.]{\textbf{Performance of domain-adaptive ReLERNN models.} Results are shown from (\textbf{A}) the background-selection and (\textbf{B}) the demography-mis-specification experiments. Each contour plot summarizes true (horizontal axis) vs. inferred (vertical axis) recombination rates ($\rho$) for the standard (gold) and domain adaptive (turquoise) models as evaluated on the held-out test dataset. The ridge along the horizontal axis of each contour is traced by a dashed line, representing the mode of the inferred value for each true value of $\rho$. Raw data underlying the contour plots are presented in Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032\#sec018}{S6} in Appendix \ref{appxC}.}
    \label{fig:DA-F4}
\end{figure}

Interestingly, \cite{adrion_predicting_2020} observed that ReLERNN was sometimes more strongly influenced by demographic mis-specification than unsupervised methods such as LDhelmet, even though it still performed better in terms of absolute error. The addition of domain adaptation appears to considerably mitigate this susceptibility to demographic mis-specification, making an excellent method even stronger.

\subsection{Efficacy of domain adaptation under various degrees of simulation mis-specification}
So far, we have examined scenarios of relatively modest simulation mis-specifica\\-tion, likely to be encountered in real applications. While domain adaptation appeared to be effective in these cases, we expect a limit to its capability when mis-specification is extreme. We therefore carried out a series of experiments to probe the performance of the \ac{dadaSIA} model under increasingly severe simulation mis-specification (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S4} in Appendix \ref{appxC}, also see \nameref{DA-methods}).

We found that \ac{dadaSIA} exhibited good performance when mis-specification was caused by genealogy inference alone or by light to moderate bottlenecks. As the bottleneck became more severe, its performance deteriorated, but even with a 5\% bottleneck, \ac{dadaSIA} still outperformed the standard model (Fig. \ref{fig:DA-F5}). To examine the limits of the method, we tested an extreme scenario with the 5\% bottleneck, background selection and an 8-fold mis-specification of recombination rate. In this case, the model performed poorly, having virtually no power to classify sweeps and large errors in its selection coefficient estimates (Fig. \ref{fig:DA-F5}). This example demonstrates that, while domain adaptation is useful over a broad range of mis-specification levels, it eventually does fail when mis-specification becomes extreme.

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{DA_figs/DA_F5.PNG}
    \caption[Performance of \acf{dadaSIA} model with different degrees of mis-specification.]{\textbf{Performance of \acf{dadaSIA} model with different degrees of mis-specification.} The performance of the model on the sweep classification task is quantified by the \ac{AUPRC} (\textbf{A}). Performance on the selection-coefficient inference task is quantified by \ac{RMSE} (\textbf{B}). In the “tree inference only” case, there is no mis-specification other than that caused by error in genealogy inference. In the “extreme” case, mis-specification consists of a 5\% bottleneck, background selection and an 8-fold mis-specification in recombination rate. See Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032\#sec018}{S4} in Appendix \ref{appxC} for illustrations of the different bottlenecks and \nameref{DA-methods} for details.}
    \label{fig:DA-F5}
\end{figure}

Does domain adaptation compromise performance at the opposite extreme, where there is little or no simulation mis-specification? To address this question, we tested the standard and domain-adaptive ReLERNN models in a setting without any simulation mis-specification. We focused here on ReLERNN, which directly uses raw genotypic data, as opposed to SIA, which always has some mis-specification due to genealogy inference error. We observed that the standard and domain-adaptive ReLERNN models performed nearly identically when no mis-specification was present, with only minor decreases in performance (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S7} in Appendix \ref{appxC}). Thus, there is perhaps some cost in using domain adaptation when it is not needed, but, at least in our case, that cost appears to be slight.

\subsection{Application of domain-adaptive \ac{SIA} to real data}
In applications to real data, the true selection coefficient is not known, so it is impossible to perform a definitive comparison of methods. Nevertheless, it can be informative to evaluate the degree to which alternative methods are concordant, especially with consideration of their relative performance in simulation studies.

Toward this end, we re-applied our \acf{dadaSIA} model to several loci in the human genome that we previously analyzed with \ac{SIA} (\cite{hejase_deep-learning_2022}), using whole-genome sequence data from the 1000 Genomes CEU population (\cite{auton_global_2015}). For the target domain, we sampled genealogies from genome-wide \acp{ARG} inferred from the individual sequences (see \nameref{DA-methods}). The putative causal loci analyzed included \acp{SNP} at the \textit{LCT} gene (\cite{bersaglieri_genetic_2004}), one of the best-studied cases of selective sweeps in the human genome; at the disease-associated genes \textit{TCF7L2} (\cite{lyssenko_mechanisms_2007}), \textit{ANKK1} (\cite{spellicy_variant_2014}) and \textit{FTO} (\cite{frayling_common_2007}); at the pigmentation genes \textit{KITLG} (\cite{sulem_genetic_2007}), \textit{ASIP} (\cite{eriksson_web-based_2010}), \textit{TYR} (\cite{sulem_genetic_2007,eriksson_web-based_2010}), \textit{OCA2} (\cite{han_genome-wide_2008,sturm_single_2008}), \textit{TYRP1} (\cite{kenny_melanesian_2012}) and \textit{TTC3} (\cite{liu_digital_2010}), which were also analyzed by \cite{stern_approximate_2019}; and at the genes \textit{MC1R} (\cite{sulem_genetic_2007,han_genome-wide_2008}) and \textit{ABCC11} (\cite{yoshiura_snp_2006}), where \ac{SIA} reported novel signals of selection.

We found that \ac{dadaSIA} generally made similar predictions to \ac{SIA} at these \acp{SNP}, but there were some notable differences. The seven loci predicted by SIA to be sweeps were also predicted by dadaSIA to be sweeps (Table \ref{tab:DA-T1}), although \ac{dadaSIA} always reported higher confidence in these predictions (with probability of neutrality, $P_{\mathrm{neu}}<10^{-2}$ in all cases) than did \ac{SIA} ($P_{\mathrm{neu}}$ up to 0.384 for \textit{TYR}). The five loci predicted by \ac{SIA} not to be sweeps were also predicted by \ac{dadaSIA} not to be sweeps ($P_{\mathrm{neu}}>0.5$). At \textit{LCT}, the strongest sweep considered, the selection coefficient ($s$) estimated by \ac{dadaSIA} remained very close to \ac{SIA}’s previous estimate of $s = 0.01$ and also close to several prior estimates (\cite{bersaglieri_genetic_2004,mathieson_estimating_2020,mathieson_fads1_2018}). In all other cases, the estimate from \ac{SIA} was somewhat revised by \ac{dadaSIA}, generally by factors of about 2-3. Importantly, in all cases, the estimates from \ac{dadaSIA} remained much closer to those from \ac{SIA} than to estimates by other methods (Table \ref{tab:DA-T1}). Together, these observations suggest that the addition of domain adaptation does not radically alter \ac{SIA}’s predictions for real data but may in some cases improve them (see \nameref{DA-discussion}).

\begin{sidewaystable}
    \centering
    \caption{Selection coefficients in the European population estimated by domain-adaptive \ac{SIA} compared to previous estimates.}
    \vspace{5mm}
    \begin{tabular}{l | l | m{5cm} l m{8cm}}
    \hline
    % https://tex.stackexchange.com/questions/269547/rowcolor-for-a-multirow
       & & \multicolumn{3}{c}{\textbf{Estimates of selection coefficient}} \\ \cline{3-5}
      \rowcolor{white} \multirow{-2}{*}{\textbf{Gene}} & \multirow{-2}{*}{\textbf{\ac{SNP}}} & \textbf{Domain-adaptive SIA} & \textbf{SIA}* & \textbf{Previous estimates} \\ \hline
      \textit{KITLG} & rs12821256 & 0.0035 & 0.0019 & 0.0161 (\cite{stern_approximate_2019}) \\
      \textit{ASIP} & rs619865 & 0.0057 & 0.0019 & 0.0974 (\cite{stern_approximate_2019}) \\
      \textit{TYR} & rs1393350 & 0.0028 & 0.0011 & 0.0112 (\cite{stern_approximate_2019}) \\
      \textit{OCA2} & rs12913832 & 0.0093 & 0.0056 & 0.002 (\cite{stern_approximate_2019}); 0.036 (\cite{wilde_direct_2014}) \\
      \textit{MC1R} & rs1805007 & 0.0027 & 0.0037 & No selection (\cite{harding_evidence_2000}) \\
      \textit{ABCC11} & rs17822931 & 0.0020 & 0.00035 & $\approx$ 0.01 in East Asian (\cite{ohashi_impact_2011}) \\
      \textit{LCT} & rs4988235 & 0.0097 & 0.010 & $\approx$ 0.01 (\cite{bersaglieri_genetic_2004,mathieson_fads1_2018,mathieson_estimating_2020}) \\
      \textit{TYRP1} & rs13289810 & $P_{\mathrm{neu}}>0.5$ & $P_{\mathrm{neu}}>0.5$ & No selection (\cite{stern_approximate_2019}) \\
      \textit{TTC3} & rs1003719 & $P_{\mathrm{neu}}>0.5$ & $P_{\mathrm{neu}}>0.5$ & No selection (\cite{stern_approximate_2019}) \\
      \textit{TCF7L2} & rs7903146 & $P_{\mathrm{neu}}>0.5$ & $P_{\mathrm{neu}}>0.5$ & N/A \\
      \textit{ANKK1} & rs1800497 & $P_{\mathrm{neu}}>0.5$ & $P_{\mathrm{neu}}>0.5$ & N/A \\
      \textit{FTO} & rs9939609 & $P_{\mathrm{neu}}>0.5$ & $P_{\mathrm{neu}}>0.5$ & N/A \\
      \hline
      \rowcolor{white} \multicolumn{5}{p{20cm}}{*The original \ac{SIA} model in \cite{hejase_deep-learning_2022} uses genealogies \textit{inferred} from simulations for training, despite the availability of ground truth genealogies.}
    \end{tabular}
    \label{tab:DA-T1}
\end{sidewaystable}

\section{Discussion} \label{DA-discussion}
Standard approaches to supervised machine learning rest on the assumption that the data they are used to analyze follow essentially the same distribution as the data used for training. In applications in population genetics, the training data are typically generated by simulation, leading to concerns about potential biases from simulation mis-specification when supervised machine-learning methods are used in place of more traditional summary-statistic- or model-based methods (\cite{caldas_inference_2022,korfmann_deep_2023}). In this article, we have shown that techniques from the “domain adaptation” literature can effectively be used to address this problem. In particular, we showed that the addition of \iac{GRL} to two recently developed deep-learning methods for population genetic analysis -- \ac{SIA} and ReLERNN -- led to clear improvements in performance on “real” data that differed in subtle but important ways from the data used to train the models. These improvements were observed both when the demographic models were mis-specified and when background selection was included in the simulations of “real” data but un-modeled in the training data.

While we observed performance improvements in all of our experiments, they were especially pronounced in the case where \ac{SIA} was used to predict specific selection coefficients, rather than simply to identify sweeps. The standard model (with training on simulated data and testing on “real” data) performed particularly poorly in this regression setting and domain adaptation produced striking improvements (Fig. \ref{fig:DA-F3}B\&D). This selection-coefficient inference problem appears to be a harder task than either sweep classification or recombination-rate inference, and the performance in this case proves to be more sensitive to simulation mis-specification (cf. Fig. \ref{fig:DA-F3}A\&C). In general, we anticipate considerable differences across population-genetic applications in the value of domain adaptation, with some applications being more sensitive to simulation mis-specification and therefore more apt to benefit from domain adaptation, and others being less so.

We also observed some interesting differences in the ways \ac{SIA} and ReLERNN responded to domain adaptation. For example, the performance gap between the “simulation benchmark” (trained and tested on simulated data) and “hypothetical true” (trained and tested on real data) models was considerably greater for \ac{SIA} than for ReLERNN (Figs. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S2C\&D, S6C\&D} in Appendix \ref{appxC}). This difference appears to be driven by \ac{ARG} inference, which is required by \ac{SIA} in the hypothetical true case but not the simulation benchmark case, and for which no analog exists for ReLERNN. For \ac{SIA}, the uncertainty about genealogies given sequence data makes the prediction task fundamentally harder in the real world (target domain) than in simulation (source domain) (Fig. \ref{fig:DA-F1}B). By contrast, ReLERNN does not depend on a similar inference task, and therefore the target and source domains are more or less symmetric. This same factor contributed to the much more dramatic drop in performance for \ac{SIA} than ReLERNN under the “standard model,” where the model is trained on simulated data and naively applied to “real” data (Figs. \ref{fig:DA-F3}B\&D, \ref{fig:DA-F4}). It is, of course, also conceivable that simulation mis-specification has more impact on selection inference than recombination rate inference, rendering the standard \ac{SIA} model less robust than the standard ReLERNN model. Regardless of the exact cause, the result is more potential for improvement from domain adaptation with \ac{SIA} than with ReLERNN (Figs. \ref{fig:DA-F3}, \ref{fig:DA-F4}, \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S2, S6} in Appendix \ref{appxC}). In effect, in \ac{SIA}, domain adaptation not only mitigates simulation mis-specification but also compensates for \ac{ARG} inference error, as directly evidenced by the observation that domain adaptation improves model performance when mis-specification is due to genealogy inference alone (Fig. \ref{fig:DA-F5}, “Tree inference only”). More broadly, we expect domain adaptation to be especially effective in applications that depend not only on the simulated data itself but also on nontrivial inferences of latent quantities that are known for simulated but not real data.

In addition, we performed a series of experiments to probe the limits of domain adaptation. As expected, the \ac{dadaSIA} model gradually lost its power as simulation mis-specification became more severe. In an extreme case where mis-specification involved demography, selection and recombination rate, the \ac{dadaSIA} model had virtually no power to classify sweeps and exhibited high error of selection coefficient inference (Fig. \ref{fig:DA-F5}). In practice, simulation models themselves are inferred from real data. With high quality data, state-of-the-art inference tools are unlikely to fail completely (e.g., by missing a 5\% bottleneck completely, or under-estimating recombination rate by an order of magnitude). We thus expect the most extreme scenario tested here to be fairly uncommon. Nevertheless, this experiment demonstrated that there are reasonable limits to the efficacy of domain adaptation. Consequently, it is important in real-world applications to begin with the best possible simulation model, before using domain adaptation to further optimize performance.

Because the accuracy of the simulation model is typically not known a priori, it is tempting to apply domain adaptation in all cases, regardless of the true degree of mis-specification. Indeed, we found that the domain-adaptive model performed very similarly to the standard model in the absence of mis-specification (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S7} in Appendix \ref{appxC}), suggesting little risk in applying the approach liberally. When the target domain is mis-specified, the domain classifier appears to “unlearn” the mis-specification, with its loss increasing steadily before plateauing where the source and target domains are no longer distinguishable. In contrast, when there is no mis-specification, the domain classifier starts with a high loss and this loss remains high (Figs. \ref{fig:DA-F2}B, \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S8} in Appendix \ref{appxC}). In this case, because the source and target domains are effectively indistinguishable, the domain classifier can never do much better than randomly guessing, leading to near-zero gradients along the domain classifier branch. In effect, the training process ignores the domain-classifier branch in this case, and improves only the feature-extractor and label-predictor portions of the model. For this reason, the domain-adaptive model behaves nearly identically to the standard model in the absence of mis-specification.

The accuracy of even the best current selection-coefficient inference methods appears limited (\cite{flagel_unreasonable_2019,torada_imagene_2019,hejase_deep-learning_2022,stern_approximate_2019}). More work is needed on models and methods for inference as well as on the problem of simulation mis-specification. Nevertheless, current methods can still be valuable in approximately characterizing the strength of selection. In our re-analysis of several loci in the 1000 Genomes CEU population, we found that \ac{dadaSIA} made similar predictions to \ac{SIA}, but it tended to exhibit higher confidence in its predictions (Table \ref{tab:DA-T1}). Considering the extensive previous work on demography inference for the CEU population, we expect that simulation mis-specification is limited in severity for this analysis, but that some mis-specification is inevitable. Given the similar performance on benchmarks of \ac{SIA} and other leading methods such as CLUES, their similar sensitivity to moderate levels of simulation mis-specification (\cite{hejase_deep-learning_2022}), and the improvements offered by domain adaptation that are demonstrated in this work, we find it likely that \ac{dadaSIA} improves on previous estimates of selection coefficients in this setting.

In a typical application of domain adaptation, the distribution shift between the source and target domains is treated as a nuisance. However, for certain population genetic questions, the gap between the simulated and real data could in principle help to reveal unmodeled evolutionary processes. We observed that the domain classifier generally tended to start with a lower loss and took more epochs to train when the mis-specification is more severe (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S9} in Appendix \ref{appxC}). It might be worthwhile, as a future endeavor, to try to identify the features driving this loss, understand their evolutionary significance, and, perhaps, incorporate them into a new set of simulations. In such a way, domain adaptation could be used to discover evolutionary processes and improve the models used for simulation.

Although our experiments were limited to background selection and demographic mis-specification, we expect that the domain adaptation framework would also be effective in addressing many other forms of simulation mis-specification, involving factors such as mutation or recombination rates, or the presence of gene conversion. Another interesting application may be to use domain adaptation to accommodate admixed populations. Each ancestry component could be modeled as a distinct target domain using a multi-target domain adaptation technique (\cite{isobe_multi-target_2021,nguyen-meidine_unsupervised_2021,roy_curriculum_2021}). It is also worth noting that our experiments considered only one, rather simple, strategy for domain adaptation. Since the \ac{GRL} was proposed, several other architectures for deep domain adaptation have achieved even better empirical performance on computer vision tasks (see: \href{https://paperswithcode.com/task/domain-adaptation}{Papers with Code}).

Our domain-adaptation approach leaves simulations unchanged and attempts to “unlearn” their mis-specification, in contrast to other strategies that aim to improve the simulations themselves. For example, the original \ac{SIA} model was trained with inferred genealogies from the simulated sequences, rather than the true genealogies used to generate the data, to mitigate the effect of genealogy inference error (\cite{hejase_deep-learning_2022}). An alternative approach is to use a \ac{GAN} to train a simulator that accurately mimics the real data (\cite{wang_automatic_2021}). These methods can require costly preprocessing steps, but they have the advantage of explicitly addressing the simulation mis-specification in an interpretable manner.

It is perhaps worth distinguishing mis-specification along the axis of inference -- that is, of target parameters such as the selection coefficient -- from mis-specification of other “nuisance” parameters (such as demographic parameters), or similarly, other unmodeled aspects of the data-generating process (such as background selection). From our observations, domain adaptation appears to be effective at addressing mis-specification of nuisance parameters or processes, at least if it is not too severe. Mis-specification of the target parameters, however, is clearly a more challenging problem. For example, it seems unlikely that domain adaptation will ever be able to “extrapolate” beyond the range of the training examples (as it fails to do in Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S5} in Appendix \ref{appxC}). Hence, it is essential in practical applications to simulate the parameter of interest from an adequately large range. Notably, \cite{burger_neural_2022} recently developed a method that addresses mis-specification in the distribution (but not the range) of a target parameter. Their method improves inference of the scaled mutation rate when regions of the parameter space are under-sampled in the training simulations by adaptively reweighing the training data, effectively improving interpolation (but not extrapolation) from the training distribution. We view these interrelated questions of how to accommodate mis-specification of both nuisance and target parameters as promising areas for future work.

Mis-specification is not only a problem in the simulation-based supervised machine learning setting explored in this work (\textit{simulation} mis-specification), but also arises in many unsupervised methods (such as maximum-likelihood or Bayesian probabilistic models). In these cases, mis-specification typically results from simplified or incorrect assumptions built into a probabilistic model (\textit{model} mis-specification, reviewed in detail by \cite{johri_recommendations_2022}). Such model mis-specification can be difficult and time-consuming to identify and address, usually calling for careful experimental design and model comparison (\cite{johri_recommendations_2022}). In some ways, the simulation mis-specification problem is more straightforward to address through fully empirical, data-driven solutions such as domain adaptation. It remains to be seen whether these empirical techniques can be used to improve probabilistic-model-based inference methods. Overall, there is rich potential for new work to address a wide variety of mis-specification challenges in population genetics, leading to improved accuracy and robustness in inference.

\section{Methods} \label{DA-methods}

\subsection{Methodological summary of unsupervised domain adaptation}

To build domain-adaptive versions of \ac{SIA} and ReLERNN, we opted for the neural network architecture proposed by \cite{ganin_unsupervised_2014}, which involved attaching a domain classifier branch via \iac{GRL} to a layer of the original neural network where a latent representation of the data is presumably obtained. For example, in \iac{CNN}, the attachment point is usually immediately after the convolutional and pooling layers, which are primarily responsible for feature extraction. One possible heuristic for picking the attachment point is to look for a “bottleneck layer” in the original network corresponding to the lowest-dimensional representation of the input. The \ac{GRL}-containing networks consist of three components–a label predictor branch, a domain classifier branch and a feature extractor common to both branches (Fig. \ref{fig:DA-F2}A\&B). During the feedforward step, when data is fed to the neural network to obtain prediction outputs in both branches, the \ac{GRL} is inactive; it simply passes along any input to the next layer. However, during backpropagation, when the gradient of the loss function with respect to the weights of the network is calculated iteratively backward from the output layer, the \ac{GRL} inverts the sign of any incoming gradient before passing it back to the previous layer. This operation has the effect of driving the feature extractor away from distinguishing the source and target domains, and consequently encourages it to extract “domain-invariant” features of the data. This effect is manifested during training as the domain-classifier loss being \textit{maximized}. We implemented the \acp{GRL} in TensorFlow (v2.4.1) using the \texttt{tf.custom\_gradient} decorator. On top of each custom \ac{GRL}, the rest of the model was built using the \texttt{tf.keras} functional API (see the \href{https://github.com/ziyimo/popgen-dom-adapt}{GitHub} repository for details).

All models were trained with the Adam optimizer using a batch size of 64. For the domain-adaptive models, training consisted of both (1) feeding labeled data from the source domain through the label predictor and obtaining a label prediction loss (cross entropy for classification task, mean squared error for regression task); and (2) feeding a mixture of unlabeled data from both the source and target domains through the domain classifier, obtaining a domain classification loss (cross entropy) (Fig. \ref{fig:DA-F2}C). In each minibatch, back-propagation from these two steps occurred simultaneously (i.e. the weights of the feature extractor were updated according to the combination of gradient from the label predictor and reversed gradient from the domain classifier). Note that the same source-domain data (but shuffled differently) were used for both steps. Training was accomplished using a custom data generator implemented with \texttt{tf.keras.utils.Sequence}. In this study, we simply assigned equal weights to the label-prediction and domain-classification loss functions (following \cite{ganin_unsupervised_2014}). Nonetheless, the relative weights of the two branches can be tuned via a hyper-parameter $\lambda$, with potential implications for performance. Intuitively, the domain classifier should be penalized more when the simulations are more mis-specified. One potential strategy is to leverage the losses and gradients of the domain classifier to guide the choice of $\lambda$. Each training epoch took around 300 s for the domain-adaptive \ac{SIA} model and around 800 s for the domain-adaptive ReLERNN model on a single NVIDIA Tesla V100 GPU. With early-stopping, the models in this study were trained on average for tens of epochs. The runtimes for domain-adaptive \ac{SIA} and ReLERNN models were therefore on par with their standard versions (on the order of hours) (\cite{adrion_predicting_2020,hejase_deep-learning_2022}).

\subsection{Setup of benchmarking experiments}

We designed four benchmarking scenarios to contextualize the performance of the domain-adaptive models (Fig. \ref{fig:DA-F1}C). \textbf{i)} In the \textit{simulation benchmark} (\textit{source-matched}) case, we tested the original model trained with source domain data on held-out samples in the source domain. This is how model benchmarks are usually run, with the test data following the same distribution as the training data. Note that for the \ac{SIA} model, the source domain consists of true genealogies and therefore both training and testing were performed with true trees. \textbf{ii)} In the \textit{hypothetical true model} (\textit{target-matched}) case, the original model was trained and tested with labeled target domain data. Here, both training and testing were performed with inferred genealogies for the \ac{SIA} model. This is a hypothetical case because it is unlikely in the evolution setting to have large quantities of labeled data from the target domain for training (i.e. real population data with known ground truth of evolutionary parameters). This case represents the performance ceiling of a standard machine learning model trained in-domain. \textbf{iii)} The \textit{standard model application} recapitulated the usual workflow of supervised machine learning methods, where the model trained with source domain simulations was applied directly to “real” data in the target domain. This was the baseline case to which we compared the domain-adaptive model. \textbf{iv)} \textit{Domain-adaptive application} of supervised machine learning models is the novel approach introduced in this study (see above and Fig. \ref{fig:DA-F1}A).

\subsection{Background selection experiment with \ac{SIA}}

To assess the robustness of \ac{dadaSIA} to background selection, we simulated labeled examples (250,000 neutral and 250,000 sweep) in the source domain under demographic equilibrium with $N_e = 10,000$ and $\mu = \rho = 1.25\times 10^{-8}$/bp/gen. The sweep simulations consisted of 100kb chromosomal segments with a hard sweep at the central nucleotide having selection coefficient $s \in [0.002, 0.01]$. Simulations were performed in SLiM 3 (\cite{haller_slim_2019,haller_tree-sequence_2019}) followed by recapitation with msprime (\cite{baumdicker_efficient_2022}), and we kept the true genealogies as source domain data. The unlabeled data in the target domain (with the exception of held-out test dataset with labels retained) were simulated in a similar fashion, albeit with a 10kb segment (“gene”) under purifying selection at the center of each 100kb chromosomal segment. All mutations in the central 10kb segment that arose during the forward stage of the simulations (in SLiM), other than the beneficial mutation in sweep simulations, followed \iac{DFE} parameterized by a gamma distribution with a mean $\bar{s} = -0.03$, a shape parameter $\alpha = 0.2$ and had dominance coefficient $h=0.25$ (\cite{boyko_assessing_2008}). We retained only the sequence data from the target domain simulations and inferred genealogies using Relate (\cite{speidel_method_2019}). The datasets were partitioned following a 90\%:2\%:8\% train-validation-test split.

\subsection{Demography mis-specification experiment with \ac{SIA}}

In a second set of simulations, we gauged whether domain adaptation also protects \ac{SIA} against demographic mis-specification. In this case, instead of specifying the degree of mis-specification a priori, we designed an end-to-end workflow that recapitulated how demographic mis-specification arises in a realistic population genetic analysis (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S1A} in Appendix \ref{appxC}). First, we simulated “real” data (in the target domain) using an assumed demography (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S1A} in Appendix \ref{appxC}, loosely based on the three-population model in \cite{campagna_selective_2022}). Similar to what one would do with actual sequence data, we then used the “real” samples to infer a demography with G-PhoCS (\cite{gronau_bayesian_2011}), pretending that the true demography and genealogies were unknown. The G-PhoCS model assumed constant population sizes between split events and a single pulse migration from population C to B, and therefore was under-parameterized. As shown in Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S1A} in Appendix \ref{appxC}, the inferred demography was consequently somewhat mis-specified. In addition to errors in population sizes, the split between B and C was inferred to be much more recent compared to the true demographic model. This mis-specified demographic model was then used to simulate labeled training data (in the source domain) for \ac{SIA}.

With the goal of using \ac{SIA} to infer selection in population B, we simulated a soft sweep site at the center of a 100kb chromosomal segment with selection coefficient $s\in [0.003, 0.02]$ and initial sweep frequency $f_{\mathrm{init}} \in [0.01, 0.1]$, under positive selection only in population B. To improve computational efficiency, simulations were performed with a hybrid approach where the neutral demographic processes were simulated first with msprime (\cite{baumdicker_efficient_2022}), followed by positive selection simulated with SLiM 3 (\cite{haller_slim_2019,haller_tree-sequence_2019}). We produced 200,000 balanced (between neutral and sweep) simulations of “real” data, 10,000 of which were randomly held out as ground-truth test data for benchmarking with their labels preserved (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S1A} in Appendix \ref{appxC}). The rest remained unlabeled. This corresponded to a train-validation-test split of 93\%:2\%:5\%. We preserved only the sequences and used Relate (\cite{speidel_method_2019}) to infer the \ac{ARG} of population B from the “real” data. \ac{SIA} works with a single population and thus the central genealogies containing only samples from population B were encoded as input to the model. For demographic inference, we randomly downsampled 10,000 5kb loci and analyzed them with G-PhoCS, keeping 4 (diploid) individuals from population A and 16 (diploid) individuals each from populations B and C. We took the median of 90,000 \acs{MCMC} samples (after 10,000 burn-in iterations) as the inferred demography (shown in Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S1A} in Appendix \ref{appxC}). The control file used to run G-PhoCS is available in the GitHub repository. We then simulated true genealogies of population B using the inferred demography, yielding 200,000 balanced samples with neutral/sweep and selection coefficient labels. All \ac{SIA} models in this study used 64 diploid samples (128 taxa).

\subsection{Running \ac{SIA} under varying degrees of simulation mis-specification}

To probe the limit of domain adaptation in mitigating simulation mis-specifica\\-tion, we performed a series of experiments that gradually increased the severity of mis-specification. In all cases, the source domain consisted of 400,000 balanced samples of \textit{true} genealogies simulated under a constant $N_e$ of 10,000. The target domain had a matching size of 400,000 balanced samples of \textit{inferred} genealogies. We used $\mu = \rho = 1.25\times 10^{-8}$/bp/gen unless otherwise specified. The datasets were partitioned following an 87.5\%:2.5\%:10\% train-validation-test split. In the “tree inference only” case, the target domain consisted of inferred genealogies simulated under a constant $N_e$ of 10,000 with no demographic mis-specification. In addition, we tested four cases with $N_e$ = 8,000, 5,000, 2,000 or 500 bottlenecks between 1,000 and 2,000 generations before the present, respectively (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S4} in Appendix \ref{appxC}). Finally, we tested an “extreme” case with the $N_e = 500$ bottleneck, a mis-specified $\rho = 1\times 10^{-7}$, as well as background selection in the central 10kb region following \iac{DFE} parameterized by a gamma distribution with a mean $\bar{s}=-0.03$, a shape parameter $\alpha = 0.2$ and a dominance coefficient $h = 0.25$.

\subsection{Updates to genealogical features and deep learning architecture for the \ac{SIA} model}

For this study, we adopted a richer encoding of genealogies than the one used previously for \ac{SIA}. Instead of simply counting the lineages remaining in the genealogy at discrete time points (\cite{hejase_deep-learning_2022}), we fully encoded the topology and branch lengths of the tree using the scheme introduced by (\cite{kim_distance_2020}). Under this scheme, a genealogy with $n$ taxa is uniquely encoded by an $(n-1) \times (n-1)$ lower-triangular matrix $\mathbf{F}$ and a weight matrix $\mathbf{W}$ of the same shape. Each cell $(i, j)$ of $\mathbf{F}$ records the lineage count between coalescent times $t_{n-j}$ and $t_{n-1-i}$, whereas each cell $(i, j)$ of $\mathbf{W}$ records the corresponding interval between coalescent times, $t_{n-j}-t_{n-1-i}$ (see Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S1B} in Appendix \ref{appxC} and \cite{kim_distance_2020} for details). In addition, we used a third matrix $\mathbf{R}$ to identify the subtree carrying the derived alleles at the site of interest, following the same logic as $\mathbf{F}$ (see Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S1B} in Appendix \ref{appxC} for an example). The $\mathbf{F}$, $\mathbf{W}$ and $\mathbf{R}$ matrices have the same shape and therefore can easily be stacked as input to a convolutional layer with three channels (Fig. \ref{fig:DA-F2}A, 128 taxa yield a $127 \times 127 \times 3$ input tensor).

Unlike the previous reductive encoding of lineage counts, the new scheme is bijective (\cite{kim_distance_2020}) and therefore contains the entirety of information in the genealogy. To utilize the improved input feature consisting of stacks of matrices, we modified the neural network architecture of \ac{SIA} and used convolutional layers (Fig. \ref{fig:DA-F2}A). The new feature encoding and \ac{CNN} architecture resulted in modest gain in performance compared to the original encoding and \ac{RNN} architecture (Fig. \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{S1C} in Appendix \ref{appxC}). In this study, both the standard and domain-adaptive \ac{SIA} models use convolutional layers with the improved feature encoding. The original \ac{SIA} \href{https://github.com/CshlSiepelLab/arg-selection}{codebase} has been updated to take advantage of the new feature encoding and model architecture as well.

\subsection{Simulation study of recombination rate inference with ReLERNN}

We conducted two sets of simulation experiments to test the same two types of mis-specification as previously described for \ac{SIA}. Each simulation consisted of 32 haploid samples of 300kb genomic segment with uniformly sampled mutation rate $\mu \sim \mathcal{U}[1.875\times 10^{-8}, 3.125\times 10^{-8}]$ and recombination rate $\rho \sim \mathcal{U}[0, 6.25\times 10^{-8}]$. To test the effect of background selection, the labeled source domain data (with true values of $\rho$) were simulated under demographic equilibrium with $N_e = 10,000$, whereas the unlabeled target domain data were simulated under the same demography, but with the central 100kb region under purifying selection, as with \ac{SIA}. To test the effect of demographic mis-specification, we conducted simulations similar to those of \cite{adrion_predicting_2020} where labeled source domain data were generated under demographic equilibrium (with $N_e = 6,000$, calculated approximately by $\frac{\hat{\theta}_W}{4\mu}$ where $\hat{\theta}_W$ was estimated from the target domain data) and unlabeled target domain data were generated under a European demography (\cite{tennessen_evolution_2012}). For each domain, 500,000 simulations were generated with SLiM 3 (background selection experiment) or msprime (demography experiment), and partitioned following an 88\%:2\%:10\% train-validation-test composition. We modified the ReLERNN model to be domain-adaptive (Fig. \ref{fig:DA-F2}B) and used the simulated data to benchmark its performance against the original version of the model.

\subsection{Application of domain-adaptive \ac{SIA} model to 1000 Genomes CEU population}

Labeled training data (source domain) for \ac{SIA} were simulated with discoal (\cite{kern_discoal_2016}) under the European demographic model from \cite{tennessen_evolution_2012}. Following \cite{hejase_deep-learning_2022}, we simulated 500,000 100-kb regions of 198 haploid sequences. The per-base per-generation mutation rate ($\mu$) and recombination rate ($\rho$) of each simulation were sampled uniformly from the interval $[1.25\times 10^{-8}, 2.5\times 10^{-8}]$; the segregating frequency of the beneficial allele ($f$) was sampled uniformly from $[0.05, 0.95]$; the selection coefficient ($s$) was sampled from an equal mixture of a uniform and a log-uniform distribution with the support $[1\times 10^{-4}, 2\times 10^{-2}]$. An additional 500,000 neutral regions were simulated to train the classification model, under the identical setup sans the positively selected site.

We curated target domain data from the 1000 Genomes CEU population to train the \ac{dadaSIA} model. The genome was first divided into 2Mb windows 1,111 of which passed three data-quality filters: \textbf{1)} contained at least 5,000 variants, \textbf{2)} at least 80\% of these variants had ancestral allele information, and \textbf{3)} at least 60\% of nucleotide sites in the window passed \textit{both} the 1000 Genomes strict accessibility mask (\cite{auton_global_2015}) and the deCODE recombination hotspot mask (standardized recombination rate $>$ 10; \cite{kong_fine-scale_2010}). In each of these 1,111 windows, we randomly sampled 1,000 variants and extracted genealogical features at those variants from Relate-inferred \acp{ARG} (\cite{speidel_method_2019}), yielding around 1 million samples that constituted the unlabeled target domain data. Finally, domain-adaptive \ac{SIA} models for classifying sweeps and inferring selection coefficients were trained as described previously and applied to a collection of loci of interest (Table \ref{tab:DA-T1}).

% \section{Supplementary material}
% Supporting information is available at \href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1011032#sec018}{\textit{PLoS Genetics} online}. All code used in this study are available at \href{https://github.com/ziyimo/popgen-dom-adapt}{GitHub}. The 1000 Genomes data are available \href{https://www.internationalgenome.org/data}{online}.
