\chapter{A deep-learning approach for inference of selective sweeps from the ancestral recombination graph}

\textit{Content of this chapter was previously uploaded to bioRxiv (2021) under the title ``SIA: Selection Inference Using the Ancestral Recombination Graph" by Hussein A. Hejase, Ziyi Mo, Leonardo Campagna and Adam Siepel. The manuscript was published in Molecular Biology and Evolution (2021) under the title ``A Deep-Learning Approach for Inference of Selective Sweeps from the Ancestral Recombination Graph". H.H. and Z.M. contributed equally to this work.}

\section{Abstract}

Detecting signals of selection from genomic data is a central problem in population genetics. Coupling the rich information in the \acf{ARG} with a powerful and scalable deep-learning framework, we developed a novel method to detect and quantify positive selection: \acf{SIA}. Built on a \ac{LSTM} architecture, a particular type of a \ac{RNN}, SIA can be trained to explicitly infer a full range of selection coefficients, as well as the allele frequency trajectory and time of selection onset. We benchmarked \ac{SIA} extensively on simulations under a European human demographic model, and found that it performs as well or better as some of the best available methods, including state-of-the-art machine-learning and \ac{ARG}-based methods. In addition, we used \ac{SIA} to estimate selection coefficients at several loci associated with human phenotypes of interest. \ac{SIA} detected novel signals of selection particular to the European (CEU) population at the \textit{MC1R} and \textit{ABCC11} loci. In addition, it recapitulated signals of selection at the \textit{LCT} locus and several pigmentation-related genes. Finally, we reanalyzed polymorphism data of a collection of recently radiated southern capuchino seedeater taxa in the genus \textit{Sporophila} to quantify the strength of selection and improved the power of our previous methods to detect partial soft sweeps. Overall, \ac{SIA} uses deep learning to leverage the \ac{ARG} and thereby provides new insight into how selective sweeps shape genomic diversity.

\section{Introduction}

The ability to accurately detect and quantify the influence of selection from genomic sequence data enables a wide variety of insights, ranging from understanding historical evolutionary events to characterizing the functional and disease relevance of observed or potential genetic variants. Adaptive evolution is driven by increases in frequency of alleles that enhance reproductive fitness. In addition, alleles experiencing such positive selection often provide insights into the functional or mechanistic basis of phenotypes of interest. Examples of genetic determinants of important phenotypic traits under selection in human populations include a family of mutations in the hemoglobin-$\beta$ cluster, which confer resistance to malaria and are at high frequencies in many populations (\cite{currat_molecular_2002,ohashi_extended_2004}), loci controlling growth factor signaling pathways that contribute to short stature in Western Central African hunter-gatherer populations (\cite{jarvis_patterns_2012,lachance_evolutionary_2012}), as well as mutations in several genes involved in immunity, hair follicle development, and skin pigmentation (\cite{sabeti_genome-wide_2007})(reviewed in \cite{sabeti_positive_2006, kelley_positive_2008,fu_selection_2013,hejase_summary_2020}).

Population genetic methods predominantly identify positive selection throu\-gh the detection of selective sweeps. As the frequency of an advantageous allele increases, linked variants in the vicinity can “hitchhike” to high frequency, leading to local reductions in genetic diversity. Previous approaches to detecting selective sweeps (such as traditional summary statistics [\cite{tajima_statistical_1989}], approximate likelihood and approximate Bayesian computation [\acsu{ABC}] methods [\cite{peter_distinguishing_2012}], or supervised \ac{ML} methods [\cite{schrider_shic_2016, kern_diploshic_2018}]) exploit the effect of genetic hitchhiking on the spatial haplotype structure and \ac{SFS}. Summary statistics have the advantage of being fast and easy to compute, but may confound the effects of selection on genetic diversity with the effects of complex demographic histories including bottlenecks, population expansions, and structured populations. Besides, they cannot easily be used to estimate the value of the selection coefficient. Approximate likelihood and \ac{ABC} methods, on the other hand, can provide an estimate of the strength of selection by aggregating multiple summary statistics (\cite{peter_distinguishing_2012}), but can be prohibitively computationally expensive when applied at a large scale. \ac{ML} methods for inferring selection can be more scalable and can capture complex nonlinear relationships among features. With the exception of a handful of recently developed methods that operate on the multiple sequence alignment itself (\cite{flagel_unreasonable_2019,torada_imagene_2019}), however, the majority of \ac{ML} approaches to selection inference solely make use of traditional summary statistics as features for prediction. In short, previous methods (including \ac{ABC} and most \ac{ML} methods) predominantly rely on low-dimensional summary statistics, which, even in combination, capture only a small portion of the information in the sequence data.

Recently, a new generation of inference methods have made it possible to go beyond summary statistics and estimate or sample a full \ac{ARG} (\cite{hudson_gene_1990,griffiths_ancestral_1996,wiuf_recombination_1999}) for a collection of sequences of interest. The \ac{ARG} is a complex data structure that summarizes the shared evolutionary history and recombination events that have occurred in a collection of DNA sequences, and therefore contains highly informative features that can potentially be leveraged to make accurate inferences about selection. The \ac{ARG} representation is interchangeable with a sequence of local genealogies along the genome and the recombination events that transform each genealogy to the next. The influence of selection on each allele can be characterized from the \ac{ARG}, based on departures from the patterns of coalescence and recombination expected under neutrality as reflected in the local genealogies. Traditional \ac{ARG} inference methods (\cite{hein_heuristic_1993,song_constructing_2005,minichiello_mapping_2006,kuhner_lamarc_2006,ofallon_acg_2013}) were restricted in accuracy and scalability, limiting the practical application of \acp{ARG}. Recent advances (\cite{rasmussen_genome-wide_2014}), however, have enabled scalable yet statistically rigorous genome-wide \ac{ARG} inference with dozens of genomes. Moreover, methods such as Relate (\cite{speidel_method_2019}) and tsinfer (\cite{kelleher_inferring_2019}) have further dramatically improved the scalability of \ac{ARG} inference to accommodate thousands or even hundreds of thousands of genomes. The latest progress in genealogical inference has paved the way for \ac{ARG}-based methods to address many different questions in population genetics (\cite{arenas_importance_2013,rasmussen_genome-wide_2014,kelleher_inferring_2019,speidel_method_2019}).

One natural way to exploit the richness of the \ac{ARG} representation in inference of selection would be to extract features from inferred \acp{ARG} and feed them into a modern supervised \ac{ML} framework. Deep-learning methods, in particular, have recently achieved unprecedented success on a variety of challenging problems, including image recognition, machine translation, and game-play (\cite{lecun_deep_2015}). Deep learning is also highly flexible, providing many opportunities for the design of novel model architectures motivated by biological knowledge. An \ac{ARG}-guided deep-learning model could potentially provide new insight into how natural selection impacts the human genome, human diseases and other phenotypes, and human evolution.

With these goals in mind, we developed a new method, called \acf{SIA}, that uses an \ac{RNN} (\cite{hochreiter_long_1997,maas_learning_2011}) to infer the selection coefficient and \ac{AF} trajectory of a variant that maps to a gene tree embedded in an \ac{ARG}. Rather than relying on traditional sequence-based summary statistics, \ac{SIA} makes use of features based on the local genealogies extracted from the \ac{ARG}. Based on these local topological features, \ac{SIA} learns to infer the selection coefficient and \ac{AF} trajectory of a beneficial variant (see Fig. \ref{fig:SIA-F1}). As described below, \ac{SIA} performs well on benchmarks and is reasonably robust to model mis-specification. Applying \ac{SIA} to data from the 1000 Genomes Northern and Western European (CEU) population, we identified new and known loci under positive selection that are associated with a variety of phenotypes and estimated selection coefficients at these loci. In addition, using \ac{SIA}, we built on our previous work (\cite{hejase_genomic_2020}) on a bird species-complex in the genus Sporophila by elucidating the strength and targets of selection at specific loci tied to a collection of rapid speciation events. Overall, \ac{SIA} is the first method that couples \ac{ARG}-based features with an \ac{ML} approach for population genetic inference.

\begin{figure}[h]
    \centering
    \includegraphics[width=\textwidth]{SIA_figs/SIA_F1.jpeg}
    \caption[A high-level framework for automating the detection of selective sweeps.]{\textbf{A high-level framework for automating the detection of selective sweeps.} We first estimate the demographic history for the population of interest, then based on the estimated demographic history, we simulate neutral regions and sweeps using the discoal simulator (\cite{kern_discoal_2016}). We proceed with \ac{ARG} inference and then extract \ac{ARG}-level statistics from each simulated region. The \ac{ARG}-level statistics are used as features for a deep-learning \ac{RNN} model. Finally, the trained model is applied to the empirical data to infer sweeps, selection coefficients, and \ac{AF} trajectories.}
    \label{fig:SIA-F1}
\end{figure}

\section{Results}
\subsection{Methodological overview}
\ac{SIA} is based on \iac{RNN} that is trained to predict selection at a genomic site from genealogical features at that site of interest and nearby sites (see \nameref{methods} for detailed descriptions; see Fig. \ref{fig:SIA-F1} for a conceptual overview of \ac{SIA}; and Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S1} online for an illustration of \ac{ARG} features and the \ac{RNN} architecture). Based on the demography of a particular population of interest, training data including genomic regions under various strengths of selection are simulated. The \ac{ARG} is then inferred from each simulated data set. \ac{ARG}-level statistics are extracted at the site under selection (or a neutral site) as features to be used as input to the deep-learning model. Specifically, we use lineage counts at a set of discrete time points as a fixed-dimension encoding of a genealogy. The encoding of the genealogy at the focal site as well as similar encodings of flanking genealogies constitute the feature vector for that site. \ac{SIA} uses \iac{LSTM} architecture, designed specifically to handle the temporal nature of the feature set. The \ac{LSTM} unrolls temporally such that the lineage counts at each time point are fed to the network iteratively. Finally, the model trained on simulations is applied to \acp{ARG} inferred from empirical data to identify sweeps, infer selection coefficients, and \ac{AF} trajectories.

\subsection{Classification of sweeps}
We first compared \ac{SIA} with several existing methods, including the Tajima’s D (\cite{tajima_statistical_1989}) and H1 (\cite{garud_recent_2015}) summary statistics, iHS (\cite{voight_map_2006}), a genealogy-based statistic (\cite{speidel_method_2019}), and a summary-statistic-based \ac{ML} method (\cite{schrider_shic_2016,kern_diploshic_2018}) (see \nameref{methods}), in the classification task of distinguishing hard sweeps from neutrally evolving regions. Our performance comparison was conducted across 16 combinations of selection coefficients and segregating allele frequencies such that the beneficial site was subjected to selection ranging from weak to strong, resulting in low to high derived allele frequencies (\acsu{DAF}s). Because a priori we expected sweep sites with lower selection coefficients and lower \acp{DAF} to be harder to detect, we performed a stratified analysis of \ac{SIA}’s performance by selection coefficient and \ac{DAF}. Figure \ref{fig:SIA-F2} reports the \acf{ROC} curves using simulations based on the CEU demographic model (\cite{tennessen_evolution_2012}) where inferred genealogies were used as input to \ac{SIA} to account for gene tree uncertainty. As expected, all methods tended to perform better in a regime with higher selection coefficients and \acp{DAF}, as indicated by increasing values of the \ac{AUROC} statistic from left to right (increasing selection) and from top to bottom (increasing \ac{DAF}). \ac{SIA} outperformed the other methods across model conditions, with a more pronounced performance advantage for sites under weaker selection and segregating at lower \acp{DAF} (Fig. \ref{fig:SIA-F2}). For each given selection coefficient, the \ac{AUROC} of the Relate tree statistic (shown in red in Fig. \ref{fig:SIA-F2}), which measures how unlikely it is that the observed expansion of the derived lineages is purely due to genetic drift, did not substantially improve as the \ac{DAF} increased. Alleles at higher frequency tend to be older and subjected to drift over longer periods, which may lead to reduced power for Relate to distinguish lineage expansion under selection from the neutral expectation. Consequently, although the \ac{ARG}-based methods \ac{SIA} and Relate both outperformed other methods at low \acp{DAF}, \ac{SIA} was alone in maintaining this advantage at higher \acp{DAF}.

\begin{figure}%[h]
    \centering
    \includegraphics[width=\textwidth]{SIA_figs/SIA_F2.jpeg}
    \caption[Classification performance of \ac{SIA} and other methods on simulated data.]{\textbf{Classification performance of \ac{SIA} and other methods on simulated data.} Sequence data were simulated under a variety of selection regimes (\textit{s}, shown horizontally) and \acp{DAF} for the beneficial mutation under selection (\textit{f}, shown vertically) (see \nameref{methods} for more details). The prediction task distinguished neutral regions and sweeps. The methods were tested on a set of 200 regions per panel (100 per class), and the \ac{ROC} curve records the \acf{TP} rate as a function of the \acf{FP} rate. The curve is obtained by varying the prediction threshold from 0 to 1 and recording for each threshold the number of regions correctly assigned (\acp{TP}) or misassigned (\acp{FP}) as positives (with prediction probability above the threshold). The performance of each method was evaluated based on the area under its \ac{ROC} curve, or \ac{AUROC} (shown in parenthesis in figure legend). Note that inferred genealogies were used as input to \ac{SIA}.}
    \label{fig:SIA-F2}
\end{figure}

In addition, we validated the ability of \ac{SIA} to classify genomic regions with additional test sets simulated under a demographic model for southern capuchinos, a group of songbirds in which we previously identified and characterized many examples of sweeps (\cite{hejase_genomic_2020}), finding a predominance of “soft” rather than “hard” sweeps (meaning that they tend to be based on standing genetic variation rather than new mutations; see \nameref{methods}). Figure \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S2} online reports the \ac{ROC} curves for the task of distinguishing partial soft sweeps from neutral regions. Despite soft sweeps being harder to detect, the classifier achieved good performance in the moderate-to-strong selection regimes ($s=0.005$ and $s=0.0075$) where the accuracy ranged between 82\% and 96\%, a substantial improvement over the previous accuracy of 56\% (\cite{hejase_genomic_2020}). \ac{SIA} performed particularly well in identifying partial soft sweeps when the site under selection was at a high segregating frequency. For example, at segregating frequencies of 0.75 and 0.9, the performance of \ac{SIA} ranged between 80\% and 96\% across a variety of selection regimes ($s=0.0025$, 0.005, and 0.0075). The performance of \ac{SIA} degraded somewhat for weak selection ($s=0.001$) with an accuracy ranging between 63\% and 74\%.

\subsection{Selection coefficient inference using true gene trees}
We assessed the performance of \ac{SIA} in correctly predicting the selection coefficient and compared it with CLUES (\cite{stern_approximate_2019}). Like \ac{SIA}, CLUES uses local genealogies based on the \ac{ARG} to infer a selection coefficient. However, CLUES calculates the likelihood of the genealogy analytically using \iac{HMM}, and does not rely on simulated training data. In addition, CLUES uses a single genealogy at the focal site, whereas \ac{SIA} additionally considers flanking trees.

We began by supplying both methods with true genealogies, in order to later disentangle the error deriving from the \ac{ARG} inference step from other sources of error (see \nameref{discussion}). We found that \ac{SIA} identified regions under neutrality with approximately no bias (median inferred $s=7.5\times 10^{-5}$; Fig. \ref{fig:SIA-F3}). Similarly, \ac{SIA} correctly inferred the selection coefficient for regions under moderate to strong selection ($s \in \{0.0025, 0.005, 0.0075, 0.01\}$) with the median inferred $s$ deviated from the true $s$ by at most 3\%. On the other hand, \ac{SIA} somewhat underestimated the selection coefficient (median inferred $s = 0.00037$) for the weak selection regime (true $s = 0.001$), likely owing to limits in the training set within that selection regime (see \nameref{discussion}). We further binned the results by segregating frequency and selection coefficient and found that, in general, the variance in estimates of $s$ for \ac{SIA} (as well as CLUES) tended to decrease as the segregating frequency of the beneficial allele increased (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S3 online}).

\begin{figure}[h]
    \centering
    \includegraphics[width=\textwidth]{SIA_figs/SIA_F3.jpeg}
    \caption[Predictions of selection coefficients on simulated regions using \ac{SIA} and CLUES based on true genealogies.]{\textbf{Predictions of selection coefficients on simulated regions using \ac{SIA} and CLUES based on true genealogies.} \textbf{(\textit{A})} The distribution of inferred selection coefficients for each method under each model condition are reported using a box plot. The box plot for each method reports these five statistics (from bottom to top): minimum, first quartile, median, third quartile, and maximum. The $y$-axis shows the inferred selection coefficient, whereas the $x$-axis shows the true selection coefficient. The dashed-black line indicates the true selection coefficient for each model condition. The simulations are based on the CEU demographic model and true genealogies were used as input to both methods. Each model condition (i.e., box plot) represents a set of 400 independent simulations. The mean ranks and variances of the distributions of inferred $s$ were compared using the Wilcoxon signed-rank test ($p_\mathrm{W}$) and the Brown–Forsythe test ($p_{\mathrm{BF}}$), respectively. \textbf{(\textit{B})} The \acf{RMSE} for each method under each model condition evaluated on 400 independent simulations.}
    \label{fig:SIA-F3}
\end{figure}

CLUES performed roughly similarly to \ac{SIA} in this experiment, but tended to slightly overestimate $s$ for the neutral regions (i.e., true $s=0$) and underestimate $s$ for the moderate to high selection regimes (i.e., true $s=$ 0.005, 0.0075, and 0.01). Under these conditions, \ac{SIA}’s median predictions of $s$ were noticeably closer to the true values (Fig. \ref{fig:SIA-F3}A). At the same time, CLUES performed slightly better than \ac{SIA} in weak selection regimes (i.e., true $s=$ 0.001 and 0.0025) (Fig. \ref{fig:SIA-F3}). Overall, \ac{SIA} (\ac{RMSE} = $9.52\times 10^{-4}$) achieved a lower error in estimating s than CLUES (\ac{RMSE} = $1.44\times 10^{-3}$), when true genealogies were used as input to both methods (Wilcoxon signed-rank test for difference in mean of squared error, $P=1.25\times 10^{-42}$). This finding potentially reflects the benefit of linkage information utilized by \ac{SIA} through the additional flanking genealogies (see \nameref{discussion}).

\subsection{Selection coefficient inference using inferred gene trees}
To account for gene-tree uncertainty, we next used \acp{ARG} inferred with Relate, which is scalable to the size of the training data set for \ac{SIA} (see \nameref{methods}), as input to \ac{SIA} and CLUES and compared their performance on CEU simulations. Using a reduced sample size of 32 haplotypes, we additionally compared \ac{SIA} with CLUES supplied with genealogies sampled using ARGweaver. Furthermore, we compared both methods with a supervised \ac{ML} method, ImaGene (see \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{Fig. S23 online}), that operates directly on an image of the alignment itself. ImaGene does not require gene trees as input and instead uses \iac{CNN} to perform dimensionality reduction of the sequence alignment, allowing for accurate and efficient classification and regression.

Overall, we found that \ac{SIA} and ImaGene outperformed CLUES in these experiments (Fig. \ref{fig:SIA-F4}). CLUES tended to underestimate selection coefficients for the moderate-to-strong selection regimes, to a greater extent compared with the case where true genealogies were used for inference (Figs. \ref{fig:SIA-F3}A and \ref{fig:SIA-F4}A). This decrease in performance of CLUES evidently derives from error at the \ac{ARG} reconstruction step. \ac{SIA}, on the other hand, appeared to be more robust to the same \ac{ARG} reconstruction error, and maintained an advantage even when CLUES was provided posterior samples of genealogies from ARGweaver (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S5 online}). ImaGene performed remarkably similarly to \ac{SIA}, given that it relies solely on the sequence alignment. \ac{SIA} exhibited lower error at neutral sites and sites with low-to-moderate values of $s$, whereas ImaGene prevailed at sites under strong selection (Fig. \ref{fig:SIA-F4}B). Nevertheless, \ac{SIA} showed a slightly smaller overall \ac{RMSE} ($2.75\times 10^{-3}$) compared with ImaGene ($2.91\times 10^{-3}$) (Wilcoxon signed-rank test, $P = 6.18\times 10^{-38}$), and in particular, \ac{SIA} produces estimates of $s$ much closer to 0 for neutral loci. Notably, in this case both \ac{SIA} and ImaGene were trained with simulations under the same uniform distribution of $s$ values (see \nameref{methods}). A different choice of training distribution could impact their performance across selection regimes (see \nameref{discussion}). Furthermore, we binned the results of these methods by both the segregating frequency and the selection coefficient (see Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S4 online}) and again found that in general they exhibit higher variance under low segregating frequency of the beneficial allele. As before, we also tested our regression framework on true and inferred gene trees of test sets simulated under the \textit{Sporophila hypoxantha} demographic model (see Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S6 online}). We found that \ac{SIA} was approximately unbiased for the moderate ($s=0.005$) and high ($s=0.01$) selection regimes but appeared to overestimate the selection coefficient for regions under weak selection ($s=$ 0.001 and 0.0025), when both true and inferred genealogies were used as input. Furthermore, \ac{SIA} appeared to overestimate the selection coefficient for neutral regions when inferred gene trees were used as input, whereas it was approximately unbiased for true gene trees.

\begin{figure}[h]
    \centering
    \includegraphics[width=\textwidth]{SIA_figs/SIA_F4.jpeg}
    \caption[Predictions of selection coefficient on simulated regions using \ac{SIA} and CLUES based on inferred genealogies, and ImaGene.]{\textbf{Predictions of selection coefficient on simulated regions using \ac{SIA} and CLUES based on inferred genealogies, and ImaGene.} \textbf{(\textit{A})} The distribution of inferred selection coefficients and \textbf{(\textit{B})} \ac{RMSE} for each method under each model condition. The simulations are based on the CEU demographic model where inferred genealogies were used as input to \ac{SIA} and CLUES, whereas sequence alignments were used as input to ImaGene. Figure layout and description are otherwise similar to Figure \ref{fig:SIA-F3}.}
    \label{fig:SIA-F4}
\end{figure}

\subsection{Performance on selection coefficient prediction with different sample sizes}
To explore the tradeoffs associated with the use of larger data sets, we examined the performance of \ac{SIA} under different sample sizes, assuming a constant-sized demographic model ($N_e = 10,000$). Figure \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S7 online} shows the error in selection coefficient inference on a held-out test set, stratified by the age of the allele (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S7A and B online}) and present-day \ac{DAF} (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S7C and D online}) at the site of interest. We observed that sites with low frequency (\ac{AF} $<0.33$) and more recent (onset $<0.2\times 2N_e$ generations) alleles experience the most significant reduction in error as sample size increases. Notably, the performance of \ac{SIA} on more ancient alleles (onset $>0.2\times 2N_e$ generations) had little to no improvement as the sample size increased from 32 to 254. These observations are in line with the expectation that having more samples improves the chance of capturing low-frequency alleles, but provides limited information about more ancient events. The reason for this age-dependency is that, looking backwards in time, most lineages coalesce rapidly and only a few survive to more ancient epochs, in a manner that depends only weakly on the sample size. It may be useful to consider these observations when choosing the sample size for use in studying selection in a particular context (see \nameref{discussion}).

\subsection{Inference of \ac{AF} trajectory}
We further adapted the deep-learning architecture of \ac{SIA} to model the \ac{AF} trajectory at a site by retaining the output of the \ac{LSTM} at each time point (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S1 online}; see \nameref{methods}). We then evaluated the performance of \ac{SIA} in the inference of the \ac{AF} trajectory using simulations under the CEU demography across a range of selection coefficients and current \acp{DAF}. \ac{SIA} was largely able to capture the expected trend of more rapidly increasing \ac{AF} under stronger selection (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S8 and S11 online}). In addition, \ac{AF} estimates by \ac{SIA} using both true and inferred genealogies were generally unbiased, although \ac{AF} at more recent time points tended to be slightly underestimated when data was simulated under weaker selection. \ac{AF} estimates also appeared to be more accurate in terms of variance for alleles under stronger selection (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S9 and S12 online}). As expected, the variance of \ac{AF} estimates tended to increase going further back in time (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S9 and S12 online}). We also observed that overall \ac{SIA} tended to produce more accurate \ac{AF} estimates than CLUES (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S9 and S10 online}).

\subsection{Model performance on simulations with mis-specified demographic models}
To evaluate the robustness of \ac{SIA} to mismatches between the demographic parameters used for simulating training data and the true underlying demography of real data, we tested the method on the selection-coefficient inference task with data sets simulated under a range of alternative parameters. Each aspect of this model mis-specification was assessed independently of the others. In particular, the mis-specified data sets contained simulations under 1) combinations of population mutation ($\theta$) and recombination ($\rho$) rates sampled beyond the range used for the training data (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S13 and S16 online}); 2) various alternative demographic scenarios (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S14, S17, and S19 online}); and 3) various effective population sizes (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S15 and S18 online}). We compared the performance of \ac{SIA} on these mis-specified data sets with that of CLUES (\cite{stern_approximate_2019}), supplying both methods with the true genealogies. We consider CLUES the “silver standard” when it comes to robustness because it is unsupervised and therefore should not be susceptible to mis-specified training data compared with supervised learning methods such as \ac{SIA}. Overall, we found that both CLUES and \ac{SIA} were reasonably robust to model mis-specification (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S13–S15 online}), although the performance of both methods inevitably declined when tested on severely mis-specified data (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S15 online}). Interestingly, \ac{SIA} tended to overestimate selection coefficient when the true $N_e$ was much smaller than that used for training, and underestimate it when the true $N_e$ was much larger, whereas CLUES did the opposite (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S15 online}). Because the CLUES likelihood model of \ac{AF} transition is parameterized by the population-scaled selection coefficient ($\alpha = 2Ns$), a larger $N_e$ likely appears to CLUES as equivalent to a higher $s$. On the other hand, features used by \ac{SIA} capture broad information of coalescence and linkage in the \ac{ARG}, and therefore can be distorted by mis-specified $N_e$ in more subtle ways (see \nameref{discussion}). Using the same mis-specified data set, we also ran \ac{SIA} with Relate-inferred genealogies and compared its performance with that of the genotyped-based deep-learning model ImaGene (\cite{flagel_unreasonable_2019,torada_imagene_2019}). In general, \ac{SIA} appeared to be more robust to model mis-specifications, achieving an overall \ac{RMSE} of 0.00362, 0.00318, and 0.00374 in the mis-specified $\theta/\rho$, demography, and $N_e$ experiments, respectively, compared with ImaGene, whose \ac{RMSE} was 0.00416, 0.00330, and 0.00462 in the corresponding experiments (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S16–S18 online}). The advantage of \ac{SIA} was particularly noticeable in cases of mis-specified demographic parameters (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S17 and S18 online}). Notably, \ac{SIA} exhibited reduced bias when working with inferred genealogies compared with true genealogies, under conditions of extremely mismatched $N_e$ (compare Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S15 and S18 online}).

\subsection{Model prediction at genomic loci of interest in CEU population}
We then applied the \ac{SIA} model to identify selective sweeps and infer selection coefficients at selected genomic loci in the 1000 Genomes CEU population. These loci included the canonical example of selection at the \textit{MCM6} gene, which regulates the neighboring \textit{LCT} gene and contributes to the lactase persistence trait (\cite{bersaglieri_genetic_2004}), the \textit{ABCC11} gene regulating earwax production, several pigmentation-related genes, as well as genes associated with obesity, diabetes and addiction (Table \ref{tab:SIA-T1}).


\begin{sidewaystable}
    
    \centering
    \caption{List of genomic loci of interest along with their \acfp{DAF}, sweep probabilities, and selection coefficients inferred by \ac{SIA} in the 1000 Genomes CEU population.}
    \vspace{5mm}
    \begin{tabular}{m{6cm} r r r r r c}
        \hline
        \textbf{Gene} & \textbf{SNP ID} & \textbf{Chr} & \textbf{Position}* & \textbf{\ac{DAF}} & $P_{\mathrm{\textbf{sweep}}}$ & \textbf{Selection coefficient (95\% CI)} \\
        \hline
        \textit{LCT} (\cite{bersaglieri_genetic_2004}) & rs4988235 & 2 & 136608646 & 0.74 & 0.999 & [0.01019, 0.01056] \\
        \textit{OCA2} (\cite{han_genome-wide_2008,sturm_single_2008}) & rs12913832 & 15 & 28365618 & 0.77 & 0.750 & [0.00539, 0.00575] \\
        \textit{MC1R} (\cite{sulem_genetic_2007,han_genome-wide_2008}) & rs1805007 & 16 & 89986117 & 0.12 & 0.949 & [0.00362, 0.00384] \\
        \textit{ABCC11} (\cite{yoshiura_snp_2006}) & rs17822931 & 16 & 48258198 & 0.13 & 0.620 & [0.00034, 0.00036] \\
        \textit{ASIP} (\cite{eriksson_web-based_2010}) & rs619865 & 20 & 33867697 & 0.12 & 0.777 & [0.00172, 0.00197] \\
        \textit{TYR} (\cite{sulem_genetic_2007,eriksson_web-based_2010}) & rs1393350 & 11 & 89011046 & 0.24 & 0.616 & [0.00085, 0.00135] \\
        \textit{KITLG} (\cite{sulem_genetic_2007}) & rs12821256 & 12 & 89328335 & 0.13 & 0.869 & [0.00183, 0.002] \\
        \textit{TYRP1} (\cite{kenny_melanesian_2012}) & rs13289810 & 9 & 12396731 & 0.37 & 0.144 & [0.00004, 0.00006] \\
        \textit{TTC3} (\cite{liu_digital_2010}) & rs1003719 & 21 & 38491095 & 0.62 & 0.011 & [0, 0] \\
        \textit{OCA2} & rs7495174 & 15 & 28344238 & 0.94 & 0.013 & [0, 0.00005] \\
        \textit{TCF7L2} (\cite{lyssenko_mechanisms_2007}) & rs7903146 & 10 & 114758349 & 0.69 & 0.035 & [0, 0] \\
        \textit{ANKK1} (\cite{spellicy_variant_2014}) & rs1800497 & 11 & 113270828 & 0.80 & 0.045 & [0, 0] \\
        \textit{FTO} (\cite{frayling_common_2007}) & rs9939609 & 16 & 53820527 & 0.56 & 0.011 & [0, 0] \\
        \hline
        \rowcolor{white} \multicolumn{7}{l}{*Genomic coordinates in GRCh37 (hg19) assembly}
    \end{tabular}
    \label{tab:SIA-T1}
\end{sidewaystable}

For \textit{LCT}, \ac{SIA} detected a strong signal of selection at the nearby \acs{SNP} that has been associated with the lactase persistence trait (rs4988235). At this \acs{SNP}, \ac{SIA} inferred a sweep probability close to 1 and a selection coefficient $>0.01$, making this one of the strongest signals of selection in the human genome. A close examination of the local genealogy at this site reveals a clear pattern indicative of a selective sweep––a burst of recent coalescence among the derived lineages (orange taxa are the lineages carrying the derived allele) is clearly visible from the tree (Fig. \ref{fig:SIA-F5}).

\begin{figure}
    \centering
    \includegraphics[scale=0.25]{SIA_figs/SIA_F5.jpeg}
    \caption[Local genealogies at six loci inferred to be under positive selection in the 1000 Genomes CEU population.]{\textbf{Local genealogies at six loci inferred to be under positive selection in the 1000 Genomes CEU population.} Gene name, RefSNP number, derived \ac{AF}, \ac{SIA}-inferred sweep probability and \ac{SIA}-inferred selection coefficient range for each locus are indicated at the top of each panel (see Table \ref{tab:SIA-T1} for more details). Taxa carrying the ancestral and derived alleles are colored in blue and orange, respectively.}
    \label{fig:SIA-F5}
\end{figure}

At a number of pigmentation genes (\cite{sulem_genetic_2007,han_genome-wide_2008,sturm_single_2008,liu_digital_2010,kenny_melanesian_2012}), \ac{SIA} detected signals of moderate selection, including \textit{MC1R} (rs1805007, $P_{\mathrm{sweep}} = 0.95$, $s \approx 0.0037$), \textit{KITLG} (rs12821256, $P_{\mathrm{sweep}} = 0.87$, $s \approx 0.0019$), \textit{ASIP} (rs619865, $P_{\mathrm{sweep}} = 0.78$, $s \approx 0.0019$), \textit{OCA2} (rs12913832, $P_{\mathrm{sweep}} = 0.75$, $s \approx 0.0056$), and \textit{TYR} (rs1393350, $P_{\mathrm{sweep}} = 0.62$, $s \approx 0.0011$). In addition, \ac{SIA} identified a weak signal of selection at a \acs{SNP} in the \textit{ABCC11} gene (rs17822931), which influences earwax and sweat production (\cite{yoshiura_snp_2006}), with a selection coefficient of around 0.00035. There are few other estimates for these genes available for comparison, but, notably, our estimate for \textit{LCT} of $s \approx 0.01$ is consistent with a previous estimate on the order of 0.01–0.1 (\cite{bersaglieri_genetic_2004}), and with recent studies of ancient DNA samples (\cite{mathieson_fads1_2018,mathieson_estimating_2020}) suggesting a value closer to 0.01. Our estimates suggest that selection at the pigmentation loci is considerably weaker than at \textit{LCT}, in contrast to previous estimates for these loci, which covered a wide range but were generally considerably larger (ranging from 0.02 to 0.1) (\cite{wilde_direct_2014}). Interestingly, CLUES estimated $s$ at the \textit{OCA2} locus to be on the order of 0.001 (roughly similar to \ac{SIA}’s estimate of 0.0056), but $s$ at the \textit{KITLG}, \textit{ASIP}, \textit{TYR} loci to be $>0.01$ (in comparison to \ac{SIA}’s considerably smaller estimates of 0.0019, 0.0019, and 0.0011) (\cite{stern_approximate_2019}). The apparent discrepancy between the estimates may be partially due to the fact that the two methods used samples from two different populations (CEU for SIA and GBR/British for CLUES).

On the other hand, SIA did not detect significant evidence of positive selection at several disease-associated loci (rs7903146/\textit{TCF7L2}, rs1800497/\textit{ANKK1}, and rs9939609/\textit{FTO}) or at several other pigmentation loci (rs13289810/\textit{TYRP1}, rs1003719/\textit{TTC3}, and rs7495174/\textit{OCA2}) (Table \ref{tab:SIA-T1}). Notably, allele frequencies at these six loci tend to be similar in African and European populations (\cite{marcus_visualizing_2017}), suggesting that they are not likely to be under strong environment-dependent positive selection, although it is possible that they have experienced very recent selective pressure that \ac{SIA} lacks the power to detect (see \nameref{discussion}). Notably, \textit{TYRP1} and \textit{TTC3} also lacked signals of selection in the CLUES analysis. Compared with the genealogies at sweep sites (Fig. \ref{fig:SIA-F5}), the trees at these putatively neutral loci lack the distinctive signature of recent bursts of coalescence among derived lineages (Fig. \ref{fig:SIA-F6}).

\begin{figure}
    \centering
    \includegraphics[scale=0.25]{SIA_figs/SIA_F6.jpeg}
    \caption[Local genealogies at six loci lacking signal of positive selection in the 1000 Genomes CEU population.]{\textbf{Local genealogies at six loci lacking signal of positive selection in the 1000 Genomes CEU population.} Gene name, RefSNP number, derived AF and probability of neutrality inferred by \ac{SIA} for each locus are indicated at the top of each panel (see Table \ref{tab:SIA-T1} for more details). Taxa carrying the ancestral and derived alleles are colored in blue and orange, respectively.}
    \label{fig:SIA-F6}
\end{figure}

\subsection{Southern capuchino species analysis}
Our previous study of southern capuchino seedeaters made use of the full \ac{ARG} and \ac{ML} to detect and characterize selective sweeps, and suggested that soft sweeps are the dominant mode of adaptation in these species (see \nameref{methods} for more details). To further characterize the targets and strengths of positive selection in these species, we applied \ac{SIA} to polymorphism data (\cite{turbek_rapid_2021}) for \textit{S. hypoxantha}, and adopted a conservative approach by reporting only sites with \ac{DAF} $\geq 0.5$, \ac{SIA}-inferred $s \geq 0.0025$, and \ac{SIA}-inferred sweep probability $P_{\mathrm{sweep}} \geq 0.99$ (see \nameref{methods}). In addition to loci near top $F_{\mathrm{ST}}$ peaks and known pigmentation-related genes (Table \ref{tab:SIA-T2}), we identified many more sites under positive selection located outside the previously scanned $F_{\mathrm{ST}}$ peaks, amounting to a total of 15,551 putative partial soft sweep sites across the 333 scanned scaffolds for \textit{S. hypoxantha}. These sites can be prioritized for further evaluation and downstream analysis. Notably, \ac{SIA} enabled us to distinguish between selection at regulatory and coding sequences, and we found that sweep loci near $F_{\mathrm{ST}}$ peaks and pigmentation genes fall mostly in noncoding regions (Table \ref{tab:SIA-T2}). We additionally surveyed all putative sweep sites identified by \ac{SIA} and found that they are indeed enriched in noncoding regions (Fisher’s exact test, $P = 6.80 \times 10^{-5}$), particularly noticeable in the “near-coding” regions (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S22 online}). Consistent with the observation that the most highly differentiated \acsp{SNP} among taxa are noncoding (\cite{campagna_repeated_2017,turbek_rapid_2021}), our finding suggests that positive selection may act on \textit{cis}-regulatory regions to drive differentiation and the subsequent speciation process. Furthermore, we examined many individual predictions in detail, considering the local trees inferred by Relate at these high-confidence predictions (Fig. \ref{fig:SIA-F7}). We found, in numerous cases, that these sweeps had distinct genealogical features, displaying evidence of a burst of coalescence events, corresponding to unusually large and young clades. Prominent examples include predictions near pigmentation-related genes \textit{ASIP}, \textit{KITL}, \textit{SLC45A2}, and \textit{TYRP1}.

\begin{table}
    \centering
    \caption{The top 25 $F_{\mathrm{ST}}$ peaks identified in \cite{hejase_genomic_2020} along with the number of partial soft sites in \textit{S. hypoxantha} identified for each scaffold using \ac{SIA}.}
    \vspace{5mm}
    \begin{tabular}{p{0.12\textwidth}p{0.22\textwidth}p{0.2\textwidth}p{0.12\textwidth}p{0.2\textwidth}}
        \hline
        \rowcolor{white} \textbf{Scaffold} & \textbf{Start position (Mb)} & \textbf{End position (Mb)} & \textbf{Length (kb)} & \textbf{No. of partial soft sites*} \\
        \hline
        59 & 5.74 & 5.86 & 120 & 11 \\
        118 & 7.16 & 7.22 & 60 & 5 \\
        252 & 0.40 & 0.54 & 140 & 3 \\
        257.1 & 21.24 & 21.78 & 540 & 26 \\
        257.2 & 24.40 & 24.84 & 440 & 43 \\
        257.3 & 28.66 & 28.96 & 300 & 10 \\
        257.4 & 31.30 & 31.38 & 80 & 8 \\
        257.5 & 5.78 & 6.20 & 420 & 25 (1) \\
        263 & 0.00 & 0.58 & 580 & 31 \\
        308 & 0.04 & 0.20 & 160 & 0 \\
        404.1 & 5.04 & 5.84 & 800 & 115 (7) \\
        404.2 & 10.76 & 10.96 & 200 & 30 \\
        412 & 3.38 & 3.62 & 240 & 15 \\
        430 & 10.98 & 11.10 & 120 & 24 \\
        567 & 2.50 & 2.80 & 300 & 0 \\
        637.1 & 6.00 & 6.32 & 320 & 2 \\
        637.2 & 6.84 & 6.92 & 80 & 4 \\
        762 & 1.65 & 1.73 & 80 & 30 \\
        766 & 1.98 & 2.10 & 120 & 1 \\
        791 & 9.90 & 9.98 & 80 & 15 \\
        1,717 & 0.92 & 0.98 & 60 & 7 \\
        3,622 & 0.96 & 1.36 & 400 & 8 \\
        1,635 & 3.71 & 3.75 & 40 & 4 \\
        1,954 & 2.8 & 2.9 & 100 & 17 \\
        579 & 0.1 & 0.16 & 60 & 0 \\
        \hline
        \rowcolor{white} \multicolumn{5}{p{0.95\textwidth}}{\textbf{Note}: To avoid cases with limited power, we focused on sites with segregating frequency $\geq 0.5$, \ac{SIA}-inferred $s > 0.0025$, and \ac{SIA}-inferred sweep probability $P_{\mathrm{sweep}} \geq 0.99$.} \\
        \rowcolor{white} \multicolumn{5}{p{0.95\textwidth}}{* The number of sweep sites in coding regions is shown in parenthesis.}
    \end{tabular}
    \label{tab:SIA-T2}
\end{table}

\begin{figure}
    \centering
    \includegraphics[scale=0.25]{SIA_figs/SIA_F7.jpeg}
    \caption[Local genealogies at six loci inferred to be under positive selection in \textit{S. hypoxantha}.]{\textbf{Local genealogies at six loci inferred to be under positive selection in \textit{S. hypoxantha}.} Contig name, position of \acs{SNP}, \ac{DAF}, \ac{SIA}-inferred selection coefficient range, and the pigmentation gene closest to the locus in question are indicated at the top of each panel. Haploid genomes carrying the ancestral and derived alleles are colored in blue and orange, respectively.}
    \label{fig:SIA-F7}
\end{figure}

\section{Discussion} \label{discussion}
The \ac{ARG} is useful for addressing a wide variety of biological questions ranging from inferring demographic parameters to estimating allele ages. \ac{SIA} exploits the particular utility of the \ac{ARG} for accurate inference of positive selection in a way that makes use of the full data set, as opposed to traditional summary statistics, which necessarily discard substantial information. Direct use of the \ac{ARG} improves upon traditional summary statistics in two key ways. First, it enables consideration of the temporal distribution of coalescence and recombination events in the history of the analyzed sequences, in contrast to traditional summary statistics that simply average over these coalescence and/or recombination events. In addition, \ac{ARG}-based methods provide better spatial resolution by separately examining individual genealogies and the recombination breakpoints between them, rather than averaging across windows containing unknown numbers of genealogies. These detailed patterns of coalescences and linkage enable the \ac{ARG}-based approaches to capture a more localized and fine-grained picture of selection (e.g., infer selection coefficient and \ac{AF} trajectory) as well as to achieve a better classification performance. This performance advantage is particularly noticeable at lower \acp{DAF} and when selection is weak, a regime where previous methods for selection inference fall short (Fig. \ref{fig:SIA-F2}).

At the same time, the supervised \ac{ML} approach sets \ac{SIA} apart from another \ac{ARG}-based method, CLUES, which approximates a full likelihood function for \acp{ARG} in the presence of selection using importance sampling and an \ac{HMM}. Although the accuracy of both \ac{SIA} and CLUES degraded when using inferred genealogies compared with true genealogies, reflecting the error and uncertainty at the \ac{ARG} inference step, \ac{SIA} appeared to be more robust to gene tree uncertainty (Figs. \ref{fig:SIA-F3} and \ref{fig:SIA-F4}). One possible reason for this observation is that CLUES effectively assumes that the selection coefficient at the focal site is conditionally independent of the flanking trees given the focal tree. This assumption should hold in the presence of fully specified genealogies, but it may make CLUES more sensitive to errors in the inferred genealogies. In other words, through its use of supervised learning, \ac{SIA} may be able to compensate for the effects of genealogy inference error on its estimation of the selection coefficient by also directly considering the flanking trees and \acs{LD}-related patterns among them. Still, the drop in accuracy observed across methods underscores the dependency of \ac{ARG}-based approaches on the \ac{ARG} inference method. For this reason, we anticipate that \ac{SIA} may benefit substantially from further improvement in \ac{ARG} inference tools (see \cite{hejase_summary_2020}).

The \ac{ARG}-based feature set distinguishes \ac{SIA} from other supervised \ac{ML} approaches for characterizing selective sweeps. \ac{SIA} uses local topological features of the \ac{ARG}, which are more informative than the \ac{SFS}- or \acs{LD}-based summary statistics employed by \ac{ML} methods such as S/HIC, SFselect, and evolBoosting. Using simulations, we demonstrated that the \ac{SIA} classifier outperformed a deep-learning method that aggregates these traditional summary statistics (Fig. \ref{fig:SIA-F2}). We also compared \ac{SIA} with ImaGene, which represents another flavor of supervised learning methods, inspired by the recent rise of \acp{CNN} for image recognition. ImaGene encodes sequence alignments as images for powerful population genetic inferences with \acp{CNN} and provides a state-of-the-art benchmark to compare against. We found that ImaGene performs remarkably well across a wide range of simulations, but \ac{SIA} does appear to be somewhat less biased and more robust to model mis-specification than ImaGene. The evolutionary information in the \ac{ARG} is implicit in the sequence alignment but some of this information may be difficult for a brute-force \ac{ML} model to discover directly.

We demonstrated that utilizing the \ac{ARG} granted \ac{SIA} considerably improved performance over deep-learning models solely employing traditional summary statistics. However, a possible drawback of an \ac{ARG}-based model is the potentially prohibitive computational overhead incurred by \ac{ARG} inference, especially as sample size grows. Picking a sample size when running \ac{SIA} involves a tradeoff between scalability (fewer samples, faster \ac{ARG} inference) and performance (more samples, slower \ac{ARG} inference). We have found that \ac{SIA} can infer selection coefficients reasonably well with as few as 16 haplotypes. Including more samples did improve performance but with a sublinear reduction in error (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S7 online}). Therefore, a sample size from a few dozen to a few hundreds—well within the capabilities of most modern \ac{ARG} inference methods—strikes a good balance between performance and scalability. Moreover, we found that larger sample sizes improved prediction performance primarily for alleles at lower frequencies but had little impact on the performance for more ancient alleles (as most lineages would have already coalesced going further back in time) (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S7 online}). This observation suggests that the choice of the sample size when applying \ac{SIA} should be guided by the biological question of interest -- ancient selection can be studied with just a handful of samples, whereas a larger sample size is better suited to detect more recent sweeps. Notably, the addition of ancient DNA samples could potentially enable selection to be inferred over much longer time scales. It should be possible to accommodate them with a relatively straightforward extension of the method.

Like other supervised learning methods, \ac{SIA} relies on simulations to generate training data. In order to apply \ac{SIA} in a particular population, a fresh set of training data tailored to that population needs to be simulated. Although it takes on the order of 100 CPU hours to simulate the training data compared with ten CPU hours to train the model (see \nameref{methods}), simulations can be easily distributed across multiple machines as each of them runs independently. Another potential drawback common to supervised methods is that they could be biased by subjective choices of simulation parameters. For example, \ac{SIA} and ImaGene cannot make accurate predictions of selection coefficients outside the range represented in the training data (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S20 online}), whereas unsupervised methods such as CLUES are not limited to a predefined range (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S21 online}). This problem could be circumvented by training on an extended range of $s$. Similarly, the tendency of \ac{SIA} to underestimate the selection coefficient for sites under weak selection (Figs. \ref{fig:SIA-F3} and \ref{fig:SIA-F4}) could be mitigated by augmenting the training set with simulations densely sampled from the weak selection regime. A more subtle issue, however, arises when the underlying generative process of the real data does not match the assumptions made for the simulations of the training data, potentially compromising the accuracy of the method when applied to real data. Thus, we tested \ac{SIA} on simulations with parameters mismatching those used in the training procedure. In general, we found that \ac{SIA} was fairly robust to alternative parameter values, although, as expected, performance did degrade somewhat under severely mis-specified models. Notably, \ac{SIA} achieved a similar level of robustness to model parameter mis-specification as the unsupervised (i.e., not relying on training data) likelihood method CLUES, yet outperformed the supervised deep-learning method ImaGene.

Applying \ac{SIA} to the CEU panel from the 1000 Genomes Project yielded several noteworthy findings at loci with known ties to phenotypes of interest. In addition to confirming the canonical signal of selective sweep at the \textit{LCT} locus, \ac{SIA} detected a novel signal of selection at a \acs{GWAS} \acs{SNP} in the \textit{MC1R} gene associated with red hair color, contrasting a previous study that could not find evidence of selection at \textit{MC1R} in the European population (\cite{harding_evidence_2000}). The derived allele at this locus segregates at around 10\% in the CEU population but is nearly absent in non-European populations (\cite{marcus_visualizing_2017}). In addition, at the \textit{MC1R} locus the Relate test statistic for selection (\cite{speidel_method_2019}), which tends to perform particularly well at low segregating frequencies (Fig. \ref{fig:SIA-F2}), falls slightly below the significance threshold of 0.05, supporting the evidence of positive selection at this locus. \ac{SIA} also detected evidence of selection at a \acs{SNP} in the \textit{ABCC11} gene reported to be the determinant of wet versus dry earwax as well as sweat production, mirroring the signal of selection previously found in the East Asian population (\cite{ohashi_impact_2011}), although selection in the CEU population appeared to be much weaker. In addition, \ac{SIA} identified selection at a few other pigmentation-related loci, yet determined previously identified \acsp{SNP} in the \textit{TYRP1} and \textit{TTC3} genes to be largely free from selection (Table \ref{tab:SIA-T1}). These results were consistent with a previous study (\cite{stern_approximate_2019}), which reported similar results for these pigmentation-related loci, albeit in a slightly different population (GBR). \ac{SIA} notably did not detect positive selection at \acs{GWAS} loci in the \textit{TCF7L2} gene associated with type-2 diabetes, the \textit{ANKK1} gene implicated in addictive behaviors, and the \textit{FTO} gene associated with obesity. Overall, this empirical study with the 1000 Genomes CEU population has illustrated how \ac{SIA} can be applied to assess natural selection at the resolution of individual sites, suggesting that it may be useful in prioritizing \acs{GWAS} variants for further scrutiny.

In our previous work on southern capuchino seedeaters (\cite{hejase_genomic_2020}) (see \nameref{methods}), we applied newly developed statistical methods for \ac{ARG} inference and \ac{ML} for the prediction of selective sweeps. We found evidence suggesting that a substantial fraction of soft sweeps is partial but had limited power to identify them (i.e., average accuracy of 56\%). \ac{SIA} considerably improved our characterization of positive selection in the southern capuchino species in two key ways. The \ac{SIA} framework performs inference of selection directly from genealogies instead of traditional summary statistics, and in doing so achieved an accuracy of up to 96\% in detecting partial soft sweeps. Consequently, we found abundant evidence of soft sweeps beyond the previously scanned $F_{\mathrm{ST}}$ peaks, and additionally were able to estimate their selection coefficients. Importantly, \ac{SIA} also took the analysis of selection beyond broad genomic windows containing sweeps to the identification of specific putative causal variants. We took advantage of this substantial improvement in genomic resolution and analyzed the distribution of these sweep sites, which revealed that positive selection on regions that likely contain \textit{cis}-regulatory elements plays a role in driving the differentiation and speciation of southern capuchino seedeaters.

Although we believe \ac{SIA} represents an important step forward in the use of the \ac{ARG} for \ac{ML}-based selection inference, there remain several possible avenues for improvement. For example, \ac{SIA} currently uses a point-estimate of the \ac{ARG}, rather than a distribution, and therefore does not explicitly take gene-tree uncertainty into account. Instead, the uncertainty of the inferred parameters is estimated with neural network dropouts (\cite{gal_dropout_2016}). The variance of parameter inference could alternatively be assessed from uncertainty in genealogy reconstruction by resampling coalescent times with Relate (\cite{speidel_method_2019}), and moreover resampling trees from the posterior distribution of \acp{ARG} with ARGweaver (\cite{rasmussen_genome-wide_2014}). Thus, it may be enlightening to compare these different approaches to analyzing uncertainty. Likewise, \ac{SIA} will greatly benefit from better algorithms for \ac{ARG} reconstruction that balance accuracy with scalability and can handle thousands of genomes. In addition, the \ac{SIA} framework was applied in the context of single-locus selective sweeps, but could be extended to study polygenic selection, by making use of summary statistics from genome-wide association studies (as in \cite{stern_disentangling_2021}) and adapting the architecture of our neural network to account for selection acting at multiple sites. Finally, the robustness of \ac{SIA} to model mis-specifications can be further improved by ensuring the simulated data is generated under a distribution that is compatible with the real target data set. We anticipate that the continual advancement in \ac{ARG} inference methods has the potential to open up many new applications for this flexible and powerful model of \ac{ARG}-based deep learning in population genetics.

\section{Materials and methods} \label{methods}

\subsection{Simulated data sets used for training and testing the \ac{SIA} model}
Training and testing data sets were generated using discoal (\cite{kern_discoal_2016}) by simulating 1,000,000 regions of length 100 kb for each model we considered (i.e., “neutral” or “hard sweep”). Aside from these regions, 2,000 were simulated for validation and 5,000 were simulated for testing. The number of sampled sequences was selected to match the number of individuals in the CEU population in the 1000 Genomes data set. Thus, a total of 198 haploid sequences were sampled. Simulations used a demographic model based on European demography (\cite{tennessen_evolution_2012}). In non-neutral simulations, selection was applied to a single focal site located in the middle of the simulated region. We sampled each of the main demographic and selection parameters from a uniform distribution: 1) mutation rate $\mu \sim \mathcal{U}(1.25\times 10^{-8}, 2.5\times 10^{-8})$; 2) recombination rate $\rho \sim \mathcal{U}(1.25\times 10^{-8}, 2.5\times 10^{-8})$; 3) selection coefficient $s \sim \mathcal{U}(0.0001, 0.02)$; and 4) segregating frequency of the site under selection $f \sim \mathcal{U}(0.01, 0.99)$. The total storage footprint for the simulations was 1.6TB. The average cost of one simulation was 0.53 s, amounting to a total of 148 CPU hours to simulate the entire training set. The cost of simulation was mitigated by parallelization across multiple compute nodes.

\subsection{ARG feature extraction} \label{fea-xtract}
For each target variant, we extracted the corresponding gene tree from the \ac{ARG}, then overlaid it with 100 discrete timepoints. These timepoints were fixed across all trees in an approximately log-uniform manner that resulted in finer discretization of more recent time scales (as in \cite{rasmussen_genome-wide_2014}). We considered biallelic sites only and assumed no recurrent mutations; thus, each mutation was assumed to occur on the branch of the tree where the ancestral allele switches to the derived. For each timepoint, we calculated the number of active ancestral and derived lineages. Furthermore, we computed the number of all active lineages (not distinguishing between ancestral and derived) at the same set of predefined timepoints in the two left- and right-flanking gene trees to account for linkage disequilibrium. We experimented with alternative numbers of flanking gene trees and found that the \ac{SIA} model with two flanking gene trees (\ac{RMSE} = 0.0027) outperforms a model with one (\ac{RMSE} = 0.0029) or no (\ac{RMSE} = 0.0030) flanking gene tree. Generally, more gene trees provide \ac{SIA} with richer linkage information and thus improve its ability to estimate the effect of positive selection on a locus. The exact threshold of diminishing returns, however, can be computationally costly to establish. We therefore opted to include two flanking gene trees while noting that the user can control this hyperparameter when running \ac{SIA}.

In the end, the \ac{ARG} feature for each locus consisted of a 600-dimensional vector, which was then used as input to an \ac{RNN}. The features for each simulated sweep region were extracted from the sweep site (by default at the center in all simulations) whereas the features for a simulated neutral region were extracted from a variant site (randomly chosen) with a predefined matched \ac{DAF}. The features for each genomic locus of interest in the CEU population were extracted from all variant sites at that locus having a \ac{DAF} of $>0.05$.

\subsection{Training \iac{RNN} to predict different modes of selection}
An \ac{RNN} was applied to the simulated training data sets to learn a classification or regression model for the task at hand. We used \iac{LSTM}, a particular form of \ac{RNN}, to accommodate the temporal nature of our features, account for long-term dependencies, and tackle the vanishing gradient problem observed in traditional \acp{RNN}. Our model had 100 timepoints with the final target output depending on the use of classification or regression. For the classification task, the final target output is a binary class label predicting whether a region is under selection or neutrality. For the regression task, the final target output is a continuous value, representing the selection coefficient or the time of selection onset. We also took a many-to-many approach to model the \ac{AF} trajectory for the site under selection. The \texttt{Keras} software was used to train and test the model. We used a two-stacked \ac{LSTM} to account for greater model complexity where the number of units in each stack was set to 100 and the hyperbolic tangent (\texttt{tanh}) was used as an activation function. The \texttt{Adam} optimization method with its default operating parameters was used to update the network weights. For the classification task, the \texttt{Softmax} activation function was applied on the final dense layer and the \texttt{binary\_crossentropy} was used to compute the cross-entropy loss between true labels and predicted labels. For the regression task, the \texttt{linear} activation function was applied on the final dense layer and the \texttt{mean\_squared\_error} loss was used. The \ac{SIA} deep-learning model took on average 7–10 h to train on a single \acs{GPU} node with 32 GB memory and four threads, whereas applying the trained model for prediction took less than a minute.

\subsection{Estimation of confidence intervals}
To turn our single-valued regression model into one capable of returning a distribution of predictions of $s$, we reused the dropout technique that is typically used during training. Dropout enables a fraction of nodes to be randomly “turned off” in a certain layer, which assists in the regularization of the model and helps prevent overfitting. We applied dropout during inference, enabling us to sample a “thinned” network to generate a sample prediction. By repeatedly sampling thinned networks, we generated a distribution of predictions and then computed confidence intervals based on this distribution (\cite{gal_dropout_2016}).

\subsection{\ac{ARG} inference}
Relate (\cite{speidel_method_2019}) (\texttt{v1.0.17}) was used for inferring \acp{ARG} underlying simulated genomic samples as well as the CEU population in the 1000 Genomes data set. For simulations under the \cite{tennessen_evolution_2012} demography, Relate was run with the true simulation parameters ($\mu$, $\rho$, and $N_e$) specified; whereas for genomic loci of the CEU population, Relate was run with a mutation rate of $2.5 \times 10^{-8}$/base/generation (\texttt{-m 2.5e-8}), a constant recombination map of $1.25 \times 10^{-8}$/base/generation and a diploid effective population size of 188,088 (\texttt{-N 376176}). The choice of mutation rate follows \cite{stern_approximate_2019} based on estimates from \cite{nachman_estimate_2000}. Although some more recent estimates have been lower (\cite{scally_revising_2012}), these differences in mutation rate are unlikely to have a major effect on our selection inference because \ac{SIA} appears to be fairly robust to mis-specification of mutation rate (Figs. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S13 and S16 online}). For simulations and genomic loci of the \textit{S. hypoxantha} population, Relate was run with $\mu = \rho = 1 \times 10^{-9}$/base/generation and a diploid $N_e$ of 130,000. The branch lengths of Relate-inferred genealogies were estimated iteratively with the \texttt{EstimatePopulationSize.sh} script in the Relate package. Specifically, population size history was inferred from the \ac{ARG}, the branch lengths are then updated for the estimated population size history and these steps are repeated until convergence. This was done for a default of five iterations (\texttt{-num\_iter 5}).

\subsection{Alternative methods for selection inference}
To benchmark the performance of \ac{SIA} for classification of sites under neutrality versus selective sweep, we ran the following methods: Tajima’s $D$ (\cite{tajima_statistical_1989}), H1 (\cite{garud_recent_2015}), iHS (\cite{voight_map_2006}), a summary statistics-based deep-learning model, and a tree-based statistic that is part of the Relate (\cite{speidel_method_2019}) program. Tajima’s $D$, H1, and iHS were calculated with the \texttt{scikit-allel} package. Haplotypes of the entire 100 kb simulated genomic segment were used for Tajima’s $D$ and H1 calculations. The unstandardized iHS was computed at every site with minor \ac{AF} $>5\%$, with respect to all other sites in the genomic segment (\texttt{min\_maf = 0.05, include\_edges = True}). iHS scores of all sites were then standardized in 50 \ac{AF} bins. Finally, the iHS score of a genomic region was taken to be the mean of the iHS scores of all of its variant sites. For the summary statistics-based deep-learning model, we made use of the summary statistics used by S/HIC (\cite{schrider_shic_2016,kern_diploshic_2018}) as features for our deep-learning architecture. These included 11 sequence-based summary statistics (see Figure 3 in \cite{schrider_supervised_2018}) which were used as features for our deep-learning model to distinguish among the two classes at hand (selective sweep vs. neutral drift). All statistics were collected along five consecutive 20-kb windows with the objective of identifying possible sweeps induced by a positively selected mutation in the third (middle) window. Some of these summary statistics corresponded to standard measures of diversity, such as ss (the number of segregating sites), $\pi$ (\cite{nei_mathematical_1979}), Tajima’s $D$ (\cite{tajima_statistical_1989}), $\theta_{\mathrm{W}}$ (\cite{watterson_number_1975}), $\theta_{\mathrm{H}}$ (\cite{fay_hitchhiking_2000}), the number of distinct haplotypes (\cite{messer_population_2013}), H1, H12, H2/H1 (\cite{garud_recent_2015}), $Z_{\mathrm{nS}}$ (\cite{kelly_test_1997}), and maximum value of $\omega$ (\cite{kim_linkage_2004}). For each of these statistics, we computed an average value for each of the five 20 kb windows for the simulated population. Finally, each summary statistic was normalized by dividing the value recorded for a given window by the sum of values across all five windows. The Relate tree-based selection test was performed with an add-on module (\texttt{DetectSelection.sh}) using the inferred genealogy with calibrated branch lengths at a site of interest, yielding a $\log_{10}P$ value for each site.

We also compared the performance of \ac{SIA} for selection coefficient inference with that of CLUES (\cite{stern_approximate_2019}) and a genotype-based \ac{CNN} framework (\cite{flagel_unreasonable_2019,torada_imagene_2019}). Selection coefficient inference from true genealogies was performed with \texttt{clues-v0} (\href{https://github.com/35ajstern/clues-v0}{last accessed November 28, 2021}). Transition probability matrices were built on a range of selection coefficients $[0, 0.05]$ at increments of 0.0001 and present-day allele frequencies $[0.01, 0.99]$ at increments of 0.01. Selection coefficient inference from Relate inferred genealogies was performed with CLUES (\href{https://github.com/35ajstern/clues}{last accessed November 28, 2021}). Branch lengths of the genealogy at the site of interest were resampled with Relate for 600 \acs{MCMC} iterations, and CLUES was run with the following arguments: \texttt{-tCutoff 10000 -burnin 100 -thin 5}. For the genotype-based \ac{CNN} model, each simulated genomic segment was preprocessed by first sorting the haplotypes and then converting the segment to a fixed-size genotype matrix. Haplotype sorting was performed by 1) calculating the pairwise Manhattan distances between haplotypes; 2) setting the haplotype with the smallest total distance to all other haplotypes as the first haplotype; and 3) sorting the remaining haplotypes in increasing distance to the first haplotype. To convert the sorted haplotypes to a fixed-size genotype matrix, centered on the middle variant of a simulated region, up to 180 variants on each side were retained. Variants beyond 180 were discarded and if there were fewer than 180, the missing variants were padded with zeros. Ancestral and derived alleles were coded with 0s and 1s, respectively. Consequently, each simulated genomic region was encoded as a $(198 \times 360)$ binary matrix, along with a real-valued vector encoding the genomic positions of the variants in the matrix. The \ac{CNN} model had a branched architecture -- one branch with five 1D convolution layers taking the genotype matrix as input and another branch with a fully connected layer taking the vector of variant positions as input. The output of the two branches was flattened, concatenated and fed into three fully connected layers, followed by a linear output layer to predict selection coefficient (Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161?login=true#supplementary-data}{S23 online}).

\subsection{Evaluation metrics}
To evaluate the performance of \ac{SIA}’s classification model and alternative methods, we computed an \ac{ROC} curve for the binary class at hand (“neutral” or “sweep”), to provide a more complete summary of the behavior of different types of errors. We further assessed the performance of \ac{SIA} and alternative methods in terms of correctly predicting the selection coefficient numerically using \acf{MAE}, \acf{RMSE}, coefficient of determination ($r^2$), and visually using a box plot that compares the simulated ground truth against the predictions by the method at hand.

\subsection{Robustness study}
We carried out an extensive analysis of the robustness of our approach, considering not only alternative demographic parameters (such as population size), but also alternative parameters for recombination rate, mutation rate, time of selection onset, and selection coefficients. In all cases, we took care to test our prediction methods under parameters well outside the range used in training.

\subsection{Analysis of CEU population in 1000 Genomes data}
We applied \ac{SIA} to infer selection coefficients and \ac{AF} trajectories in the 1000 Genomes (\cite{auton_global_2015}) CEU population at 13 genomic loci with known association to phenotypes, some of which were previously identified as likely targets of positive selection (Table \ref{tab:SIA-T1}). For each gene of interest, the \ac{ARG} was inferred with Relate from \acsp{SNP} within a 2-Mb window centered at the gene. Once the \ac{ARG} was inferred, only \acsp{SNP} with valid ancestral allele (``\texttt{AA}" \texttt{INFO} field in the \texttt{vcf} file) were retained for downstream analysis. Following the aforementioned protocol (see \nameref{fea-xtract}), features at all variant sites in the 2 Mb window above \iac{DAF} threshold of 0.05 were extracted. Lastly, the \ac{SIA} model was applied to classify neutrality versus selection, and infer selection coefficient and \ac{AF} trajectory at each site.

\subsection{Localizing sweeps in southern capuchino seedeaters}
We recently applied a combination of \ac{ARG} inference and \ac{ML} methods for identifying selective sweeps to study previously identified “islands of differentiation” in southern capuchino seedeaters and distinguish among possible evolutionary scenarios leading to their formation (\cite{hejase_genomic_2020}). Taking advantage of its improved power and genomic resolution, we applied \ac{SIA} to sequence data for the species for which we have the most samples, \textit{S.hypoxantha}. We simulated training (250,000 neutral; 250,000 soft sweeps), validation (1000 neutral; 1000 soft sweeps), and testing (2,500 neutral; 2,500 soft sweeps) data sets for \ac{SIA} under a demographic model inferred by G-PhoCS (\cite{campagna_distinguishing_2015}). Simulations were performed using discoal with the following parameters: 1) mutation rate $\mu = 1\times 10^{-9}$; 2) recombination rate $\rho = 1\times 10^{-9}$; 3) derived $N_e = 130,000$; 4) root divergence time = 1,850,000 generations ago; 5) root $N_e = 1,450,000$; 6) ancestral divergence time = 44,000 generations ago; 7) ancestral $N_e = 14,380,000$; 8) selection coefficient $s \sim \mathcal{U}(0.001, 0.02)$; 9) initial frequency at which selection starts acting on the allele $f_{\mathrm{init}} \sim \mathcal{U}(0.01, 0.05)$; and 10) segregating frequency of the site under selection $f \sim \mathcal{U}(0.25, 0.99)$. A total of 56 haploid sequences were sampled from each simulation, matching the number of \textit{S. hypoxantha} individuals (28) in the real data. The \ac{SIA} model for \textit{S. hypoxantha} was built, trained and evaluated in an otherwise similar fashion to that for the CEU population as outlined above.

Using a subset of polymorphism data from \cite{turbek_rapid_2021} of 28 \textit{S. hypoxantha} and 2 \textit{S. minuta} individuals, we applied our trained model to localize selective sweeps in \textit{S. hypoxantha} on 19 scaffolds that contain top $F_{\mathrm{ST}}$ peaks in at least one pairwise species comparison (\cite{campagna_repeated_2017}) and/or harbor known pigmentation-related genes such as \textit{ASIP} (located on scaffold 252; induces melanocytes to synthesize pheomelanin instead of eumelanin), \textit{KITL} (located on scaffold 412; stimulates melanocyte proliferation), \textit{SLC45A2} (located on scaffold 404; transports substances needed for melanin synthesis), and \textit{CAMK2D} (located on scaffold 1717; cell communication), as well as 316 scaffolds that 1) are longer than 100 kb; 2) contain more than 1,000 variants; and 3) where more than 95\% of sites have a consensus ancestral allele, as determined by four identical haplotypes for two individuals from the outgroup species \textit{S. minuta}. The \ac{ARG} was inferred with Relate for each scaffold independently. Once the \ac{ARG} was inferred, the \ac{SIA} model was applied to sites with consensus ancestral allele for classification and selection coefficient inference.

\section{Supplementary material}

Supplementary figures are available at \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{\textit{Molecular Biology and Evolution} online}. The simulation scripts and code for building and training the \ac{SIA} model are publicly available on \href{https://github.com/CshlSiepelLab/arg-selection}{GitHub}.