\chapter{A deep-learning approach for inference of selective sweeps from the ancestral recombination graph}

\textit{Content of this chapter was previously uploaded to bioRxiv (2021) under the title "SIA: Selection Inference Using the Ancestral Recombination Graph" by Hussein A. Hejase, Ziyi Mo, Leonardo Campagna and Adam Siepel. The manuscript was published in Molecular Biology and Evolution (2021) under the title "A Deep-Learning Approach for Inference of Selective Sweeps from the Ancestral Recombination Graph". H.H. and Z.M. contributed equally to this work.}

\section{Abstract}

Detecting signals of selection from genomic data is a central problem in population genetics. Coupling the rich information in the \acf{ARG} with a powerful and scalable deep-learning framework, we developed a novel method to detect and quantify positive selection: \acf{SIA}. Built on a \ac{LSTM} architecture, a particular type of a \ac{RNN}, SIA can be trained to explicitly infer a full range of selection coefficients, as well as the allele frequency trajectory and time of selection onset. We benchmarked \ac{SIA} extensively on simulations under a European human demographic model, and found that it performs as well or better as some of the best available methods, including state-of-the-art machine-learning and \ac{ARG}-based methods. In addition, we used \ac{SIA} to estimate selection coefficients at several loci associated with human phenotypes of interest. \ac{SIA} detected novel signals of selection particular to the European (CEU) population at the \textit{MC1R} and \textit{ABCC11} loci. In addition, it recapitulated signals of selection at the \textit{LCT} locus and several pigmentation-related genes. Finally, we reanalyzed polymorphism data of a collection of recently radiated southern capuchino seedeater taxa in the genus \textit{Sporophila} to quantify the strength of selection and improved the power of our previous methods to detect partial soft sweeps. Overall, \ac{SIA} uses deep learning to leverage the \ac{ARG} and thereby provides new insight into how selective sweeps shape genomic diversity.

\section{Introduction}

The ability to accurately detect and quantify the influence of selection from genomic sequence data enables a wide variety of insights, ranging from understanding historical evolutionary events to characterizing the functional and disease relevance of observed or potential genetic variants. Adaptive evolution is driven by increases in frequency of alleles that enhance reproductive fitness. In addition, alleles experiencing such positive selection often provide insights into the functional or mechanistic basis of phenotypes of interest. Examples of genetic determinants of important phenotypic traits under selection in human populations include a family of mutations in the hemoglobin-$\beta$ cluster, which confer resistance to malaria and are at high frequencies in many populations (\cite{currat_molecular_2002,ohashi_extended_2004}), loci controlling growth factor signaling pathways that contribute to short stature in Western Central African hunter-gatherer populations (\cite{jarvis_patterns_2012,lachance_evolutionary_2012}), as well as mutations in several genes involved in immunity, hair follicle development, and skin pigmentation (\cite{sabeti_genome-wide_2007})(reviewed in \cite{sabeti_positive_2006, kelley_positive_2008,fu_selection_2013,hejase_summary_2020}).

Population genetic methods predominantly identify positive selection throu\-gh the detection of selective sweeps. As the frequency of an advantageous allele increases, linked variants in the vicinity can “hitchhike” to high frequency, leading to local reductions in genetic diversity. Previous approaches to detecting selective sweeps (such as traditional summary statistics [\cite{tajima_statistical_1989}], approximate likelihood and approximate Bayesian computation [\acsu{ABC}] methods [\cite{peter_distinguishing_2012}], or supervised \ac{ML} methods [\cite{schrider_shic_2016, kern_diploshic_2018}]) exploit the effect of genetic hitchhiking on the spatial haplotype structure and \ac{SFS}. Summary statistics have the advantage of being fast and easy to compute, but may confound the effects of selection on genetic diversity with the effects of complex demographic histories including bottlenecks, population expansions, and structured populations. Besides, they cannot easily be used to estimate the value of the selection coefficient. Approximate likelihood and \ac{ABC} methods, on the other hand, can provide an estimate of the strength of selection by aggregating multiple summary statistics (\cite{peter_distinguishing_2012}), but can be prohibitively computationally expensive when applied at a large scale. \ac{ML} methods for inferring selection can be more scalable and can capture complex nonlinear relationships among features. With the exception of a handful of recently developed methods that operate on the multiple sequence alignment itself (\cite{flagel_unreasonable_2019,torada_imagene_2019}), however, the majority of \ac{ML} approaches to selection inference solely make use of traditional summary statistics as features for prediction. In short, previous methods (including \ac{ABC} and most \ac{ML} methods) predominantly rely on low-dimensional summary statistics, which, even in combination, capture only a small portion of the information in the sequence data.

Recently, a new generation of inference methods have made it possible to go beyond summary statistics and estimate or sample a full \ac{ARG} (\cite{hudson_gene_1990,griffiths_ancestral_1996,wiuf_recombination_1999}) for a collection of sequences of interest. The \ac{ARG} is a complex data structure that summarizes the shared evolutionary history and recombination events that have occurred in a collection of DNA sequences, and therefore contains highly informative features that can potentially be leveraged to make accurate inferences about selection. The \ac{ARG} representation is interchangeable with a sequence of local genealogies along the genome and the recombination events that transform each genealogy to the next. The influence of selection on each allele can be characterized from the \ac{ARG}, based on departures from the patterns of coalescence and recombination expected under neutrality as reflected in the local genealogies. Traditional \ac{ARG} inference methods (\cite{hein_heuristic_1993,song_constructing_2005,minichiello_mapping_2006,kuhner_lamarc_2006,ofallon_acg_2013}) were restricted in accuracy and scalability, limiting the practical application of \acp{ARG}. Recent advances (\cite{rasmussen_genome-wide_2014}), however, have enabled scalable yet statistically rigorous genome-wide \ac{ARG} inference with dozens of genomes. Moreover, methods such as Relate (\cite{speidel_method_2019}) and tsinfer (\cite{kelleher_inferring_2019}) have further dramatically improved the scalability of \ac{ARG} inference to accommodate thousands or even hundreds of thousands of genomes. The latest progress in genealogical inference has paved the way for \ac{ARG}-based methods to address many different questions in population genetics (\cite{arenas_importance_2013,rasmussen_genome-wide_2014,kelleher_inferring_2019,speidel_method_2019}).

One natural way to exploit the richness of the \ac{ARG} representation in inference of selection would be to extract features from inferred \acp{ARG} and feed them into a modern supervised \ac{ML} framework. Deep-learning methods, in particular, have recently achieved unprecedented success on a variety of challenging problems, including image recognition, machine translation, and game-play (\cite{lecun_deep_2015}). Deep learning is also highly flexible, providing many opportunities for the design of novel model architectures motivated by biological knowledge. An \ac{ARG}-guided deep-learning model could potentially provide new insight into how natural selection impacts the human genome, human diseases and other phenotypes, and human evolution.

With these goals in mind, we developed a new method, called \acf{SIA}, that uses an \ac{RNN} (\cite{hochreiter_long_1997,maas_learning_2011}) to infer the selection coefficient and \ac{AF} trajectory of a variant that maps to a gene tree embedded in an \ac{ARG}. Rather than relying on traditional sequence-based summary statistics, \ac{SIA} makes use of features based on the local genealogies extracted from the \ac{ARG}. Based on these local topological features, \ac{SIA} learns to infer the selection coefficient and \ac{AF} trajectory of a beneficial variant (see Fig. \ref{fig:SIA-F1}). As described below, \ac{SIA} performs well on benchmarks and is reasonably robust to model mis-specification. Applying \ac{SIA} to data from the 1000 Genomes Northern and Western European (CEU) population, we identified new and known loci under positive selection that are associated with a variety of phenotypes and estimated selection coefficients at these loci. In addition, using \ac{SIA}, we built on our previous work (\cite{hejase_genomic_2020}) on a bird species-complex in the genus Sporophila by elucidating the strength and targets of selection at specific loci tied to a collection of rapid speciation events. Overall, \ac{SIA} is the first method that couples \ac{ARG}-based features with an \ac{ML} approach for population genetic inference.

\begin{figure}[h]
    \centering
    \includegraphics[width=\textwidth]{SIA_figs/SIA_F1.jpeg}
    \caption[A high-level framework for automating the detection of selective sweeps.]{\textbf{A high-level framework for automating the detection of selective sweeps.} We first estimate the demographic history for the population of interest, then based on the estimated demographic history, we simulate neutral regions and sweeps using the discoal simulator (\cite{kern_discoal_2016}). We proceed with \ac{ARG} inference and then extract \ac{ARG}-level statistics from each simulated region. The \ac{ARG}-level statistics are used as features for a deep-learning \ac{RNN} model. Finally, the trained model is applied to the empirical data to infer sweeps, selection coefficients, and \ac{AF} trajectories.}
    \label{fig:SIA-F1}
\end{figure}

\section{Results}
\subsection{Methodological Overview}
\ac{SIA} is based on \iac{RNN} that is trained to predict selection at a genomic site from genealogical features at that site of interest and nearby sites (see \nameref{methods} for detailed descriptions; see Fig. \ref{fig:SIA-F1} for a conceptual overview of \ac{SIA}; and Fig. \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S1} online for an illustration of \ac{ARG} features and the \ac{RNN} architecture). Based on the demography of a particular population of interest, training data including genomic regions under various strengths of selection are simulated. The \ac{ARG} is then inferred from each simulated data set. \ac{ARG}-level statistics are extracted at the site under selection (or a neutral site) as features to be used as input to the deep-learning model. Specifically, we use lineage counts at a set of discrete time points as a fixed-dimension encoding of a genealogy. The encoding of the genealogy at the focal site as well as similar encodings of flanking genealogies constitute the feature vector for that site. \ac{SIA} uses \iac{LSTM} architecture, designed specifically to handle the temporal nature of the feature set. The \ac{LSTM} unrolls temporally such that the lineage counts at each time point are fed to the network iteratively. Finally, the model trained on simulations is applied to \acp{ARG} inferred from empirical data to identify sweeps, infer selection coefficients, and \ac{AF} trajectories.

\subsection{Classification of Sweeps}
We first compared \ac{SIA} with several existing methods, including the Tajima’s D (\cite{tajima_statistical_1989}) and H1 (\cite{garud_recent_2015}) summary statistics, iHS (\cite{voight_map_2006}), a genealogy-based statistic (\cite{speidel_method_2019}), and a summary-statistic-based \ac{ML} method (\cite{schrider_shic_2016,kern_diploshic_2018}) (see \nameref{methods}), in the classification task of distinguishing hard sweeps from neutrally evolving regions. Our performance comparison was conducted across 16 combinations of selection coefficients and segregating allele frequencies such that the beneficial site was subjected to selection ranging from weak to strong, resulting in low to high derived allele frequencies (\acsu{DAF}s). Because a priori we expected sweep sites with lower selection coefficients and lower \acp{DAF} to be harder to detect, we performed a stratified analysis of \ac{SIA}’s performance by selection coefficient and \ac{DAF}. Figure \ref{fig:SIA-F2} reports the \acf{ROC} curves using simulations based on the CEU demographic model (\cite{tennessen_evolution_2012}) where inferred genealogies were used as input to \ac{SIA} to account for gene tree uncertainty. As expected, all methods tended to perform better in a regime with higher selection coefficients and \acp{DAF}, as indicated by increasing values of the \ac{AUROC} statistic from left to right (increasing selection) and from top to bottom (increasing \ac{DAF}). \ac{SIA} outperformed the other methods across model conditions, with a more pronounced performance advantage for sites under weaker selection and segregating at lower \acp{DAF} (Fig. \ref{fig:SIA-F2}). For each given selection coefficient, the \ac{AUROC} of the Relate tree statistic (shown in red in Fig. \ref{fig:SIA-F2}), which measures how unlikely it is that the observed expansion of the derived lineages is purely due to genetic drift, did not substantially improve as the \ac{DAF} increased. Alleles at higher frequency tend to be older and subjected to drift over longer periods, which may lead to reduced power for Relate to distinguish lineage expansion under selection from the neutral expectation. Consequently, although the \ac{ARG}-based methods \ac{SIA} and Relate both outperformed other methods at low \acp{DAF}, \ac{SIA} was alone in maintaining this advantage at higher \acp{DAF}.

\begin{figure}%[h]
    \centering
    \includegraphics[width=\textwidth]{SIA_figs/SIA_F2.jpeg}
    \caption[Classification performance of \ac{SIA} and other methods on simulated data.]{\textbf{Classification performance of \ac{SIA} and other methods on simulated data.} Sequence data were simulated under a variety of selection regimes (\textit{s}, shown horizontally) and \acp{DAF} for the beneficial mutation under selection (\textit{f}, shown vertically) (see \nameref{methods} for more details). The prediction task distinguished neutral regions and sweeps. The methods were tested on a set of 200 regions per panel (100 per class), and the \ac{ROC} curve records the \acf{TP} rate as a function of the \acf{FP} rate. The curve is obtained by varying the prediction threshold from 0 to 1 and recording for each threshold the number of regions correctly assigned (\acp{TP}) or misassigned (\acp{FP}) as positives (with prediction probability above the threshold). The performance of each method was evaluated based on the area under its \ac{ROC} curve, or \ac{AUROC} (shown in parenthesis in figure legend). Note that inferred genealogies were used as input to \ac{SIA}.}
    \label{fig:SIA-F2}
\end{figure}

In addition, we validated the ability of \ac{SIA} to classify genomic regions with additional test sets simulated under a demographic model for southern capuchinos, a group of songbirds in which we previously identified and characterized many examples of sweeps (\cite{hejase_genomic_2020}), finding a predominance of “soft” rather than “hard” sweeps (meaning that they tend to be based on standing genetic variation rather than new mutations; see \nameref{methods}). Figure \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{S2} online reports the \ac{ROC} curves for the task of distinguishing partial soft sweeps from neutral regions. Despite soft sweeps being harder to detect, the classifier achieved good performance in the moderate-to-strong selection regimes ($s=0.005$ and $s=0.0075$) where the accuracy ranged between 82\% and 96\%, a substantial improvement over the previous accuracy of 56\% (\cite{hejase_genomic_2020}). \ac{SIA} performed particularly well in identifying partial soft sweeps when the site under selection was at a high segregating frequency. For example, at segregating frequencies of 0.75 and 0.9, the performance of \ac{SIA} ranged between 80\% and 96\% across a variety of selection regimes ($s=0.0025$, 0.005, and 0.0075). The performance of \ac{SIA} degraded somewhat for weak selection ($s=0.001$) with an accuracy ranging between 63\% and 74\%.

\subsection{Selection Coefficient Inference Using True Gene Trees}
We assessed the performance of \ac{SIA} in correctly predicting the selection coefficient and compared it with CLUES (\cite{stern_approximate_2019}). Like \ac{SIA}, CLUES uses local genealogies based on the \ac{ARG} to infer a selection coefficient. However, CLUES calculates the likelihood of the genealogy analytically using \iac{HMM}, and does not rely on simulated training data. In addition, CLUES uses a single genealogy at the focal site, whereas \ac{SIA} additionally considers flanking trees.

\section{Discussion}

\section{Materials and Methods} \label{methods}

\section{Supplementary Material}

Supplementaty figures are available at \href{https://academic.oup.com/mbe/article/39/1/msab332/6433161#supplementary-data}{\textit{Molecular Biology and Evolution} online}. The simulation scripts and code for building and training the \ac{SIA} model are publicly available on \href{https://github.com/CshlSiepelLab/arg-selection}{GitHub}.